---
title: "TERRE MatrixEQTL cis-mQTL Analysis"
output: html_notebook
---

The manuscript "A Parkinsonâ€™s disease genetic risk score associates with blood DNAm on chromosome 17" explores how a genetic risk score (GRS) for Parkinson's disease (PD) associates with blood DNA methylation in the TERRE study of French agricultural workers, and evaluates GRS associations with DNAm in each sex as well as their sensitivity to non-genetic factors. This was originally spearheaded by Will Casazza and was posted as a pre-print on medRxiv in 2023. The current revision (2024, led by Sam Schaffner) aims to address previous reviewer feedback and prepare the manuscript for re-submission.


This code first prepares genetic, methylation, and meta data for input into MatrixEQTL to calculate mQTLs in TERRE, then examines (1) which PD-associated SNPs were cis-mQTLs in TERRE, and (2) which GRS-associated CpGs from TERRE had an associated cis-mQTL.

Last updated Jan 20, 2025 by Samantha Schaffner: Identifying SNPs in LD to GRS SNPs that are mQTL for top CpGs

Based on code by Will Casazza

```{r setup, include=FALSE, eval=F}
library(tidyverse)
library(data.table)
library(minfi)
```

# Read in TERRE data
DNA methylation (updated to 2022-processed), genotype, and meta data (with 2022 cell type PCs).
```{r terre meta data, eval=F}
terre_ancestry <- fread("~/SHARE_DECIPHER/pd_grs_ewas_integration/raw_data.geno.maf.mind.sex_check.het_filter.ibd_filter.eigenvec")

terre_meta <- fread("/home1/NEURO/SHARE_DECIPHER/terre_meta_master.csv")
#terre_ctp <- fread("/home1/NEURO/SHARE_DECIPHER/processed_DNAm_data/2021/TERRE_processed_2021/meta_TERRE_funnorm_robCTP.txt")
terre_ctp <- read.csv("~/PD_GRS/meta_for_GRS_EWAS.csv") # with cell type PCs from 2022 DNAm data

terre_meta$IID <- gsub(".*_(PAE.*)", "\\1", terre_meta$IID)
terre_meta <- terre_meta %>%
  right_join(terre_ancestry, by = c("IID" = "V2")) %>%
  left_join(terre_ctp[, c("patient", "CTP_PC1", "CTP_PC2", "CTP_PC3", "CTP_PC4", "CTP_PC5", "CTP_PC6")], by = "patient")
colnames(terre_meta) #contains cell type and genotype PCs, FID, IID, and other meta variables
#245 observations

# Loading in DNAm data
# Update from previous script that used 2021-processed data
load("~/SHARE_DECIPHER/processed_DNAm_data/2022/TERRE_processed_2022/1-TERRE_RG_filtered.RData")
betas <- getBeta(PD_RG_filtered)
rm(PD_RG_filtered)

genotype <- fread("~/SHARE_DECIPHER/pd_grs_ewas_integration/TERRE_QC/raw_data.imputed.r2_30.maf_mismatch.traw")
dim(genotype)
#[1] 8354188     251
```

## Match Samples
```{r match samples, eval=F}
colnames(genotype) <- gsub(".*_(PAE.*)", "\\1", colnames(genotype))
colnames(genotype)
setcolorder(genotype, neworder = c(colnames(genotype)[1:6], terre_meta$IID))
dim(betas <- betas[, colnames(betas) %in% terre_meta$patient]) #803777    219

# new - match terre meta to betas (219 betas, 245 terre meta)
dim(terre_meta <- terre_meta[match(colnames(betas),terre_meta$patient),]) #219  96

# rename betas sample names with IID
colnames(betas) <- terre_meta$IID[match(colnames(betas), terre_meta$patient)]
all(colnames(betas)==terre_meta$IID) #TRUE

# match genotype and betas data
geno_df <- data.frame(genotype)
ncol(geno_df <- cbind(geno_df[,1:6], geno_df[,match(colnames(betas),colnames(geno_df))])) #225
all(colnames(betas) == colnames(geno_df)[-c(1:6)]) #TRUE
```

## Compute DNAm PCs
```{r dnam pcs, eval=F}
methy_PC <- prcomp(t(betas), center = T, rank. = 20)
```

## Write Covariates

Covariates used for mQTL analysis typically include genotype PCs, sex, methylation PCs, and sometimes methylation batch and cell type variables. Example references:

![GODMC](https://github.com/MRCIEU/godmc/wiki/Phenotype-and-covariate-data): Sex, age, DNAm chip (and other batch variables), height, BMI, cell counts, genotype PCs (created by pipeline), methylation PCs (created by pipeline)

![Oliva et al., Nat Genet 2023](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10249665/): Sex, 5 genotype PCs, 2 genotype batch variables

![Li et al., Clin Epigenetics 2022](https://clinicalepigeneticsjournal.biomedcentral.com/articles/10.1186/s13148-022-01385-6): Sex, 5 genotype PCs, DNAm batch variables

![Ohlei et al., medRxiv 2023](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC10775408/): Sex, 10 genotype PCs, 5 DNAm PCs, sample processing date


The code previously written by Will here included 3 genotype PCs, age, sex, head trauma, cell type PCs, and PD status. There was also an analysis with 10 DNAm PCs.

Here I will include the same covariates minus head trauma and DNAm PCs and plus DNAm plate and row variables. This will match the variables adjusted for in the updated EWAS. Not sure why head trauma was previously included here - check with Will.


```{r write covars, eval=F}
# data need to be in numeric format

terre_covar <- terre_meta[,c("V3", "V4", "V5", "PD", "age", "men", "CTP_PC1", "CTP_PC2", "CTP_PC3", "CTP_PC4", "CTP_PC5","CTP_PC6","SentrixPosition_A","plate")]
str(terre_covar)
terre_covar$SentrixPosition_A <- as.integer(as.factor(terre_covar$SentrixPosition_A))
summary(as.factor(terre_covar$SentrixPosition_A))
# 1  2  3  4  5  6  7  8 
#33 31 30 22 22 34 27 20 

write_delim(t(cbind(terre_covar)) %>% as.data.frame() %>% rownames_to_column("id"), "~/PD_GRS/mQTL/covariates_CTP_PD.txt")

write_delim(t(cbind(terre_covar[terre_covar$men==1,-("men")])) %>% as.data.frame() %>% rownames_to_column("id"), "~/PD_GRS/mQTL/male_covariates_CTP_PD.txt")

write_delim(t(cbind(terre_covar[terre_covar$men==0,-("men")])) %>% as.data.frame() %>% rownames_to_column("id"), "~/PD_GRS/mQTL/female_covariates_CTP_PD.txt")
```

## Write genotype and DNAm data
```{r write geno dnam, eval=F}
#methy_annot <- fread("~/MethylationEPIC_v-1-0_B4.csv", skip = 7)
load("~/kobor_shared_coding_resource/EPIC_fdat.RData")

# SNP POS
geno_df$CHR <- paste0("chr", geno_df$CHR)
write_delim(geno_df[, c("SNP", "CHR", "POS")], "~/PD_GRS/mQTL/snp_pos.txt")

# SNPs
geno_vars <- colnames(geno_df)[-c(1, 3:6)] #220 (219)
geno_vars_m <- geno_vars[-(1 + which(terre_meta$men == 0))] #120 (119)
geno_vars_f <- geno_vars[-(1 + which(terre_meta$men == 1))] #101 (100)
fwrite(geno_df[, geno_vars], "~/PD_GRS/mQTL/all_imputed_matrixeQTL.txt", sep = " ", quote = F)
fwrite(geno_df[, geno_vars_f], "~/PD_GRS/mQTL/female_all_imputed_matrixeQTL.txt", sep = " ", quote = F)
fwrite(geno_df[, geno_vars_m], "~/PD_GRS/mQTL/male_all_imputed_matrixeQTL.txt", sep = " ", quote = F)

# Methy POS
pos_dat <- data.frame(geneid=fData_EPIC$TargetID, chr=paste0("chr", fData_EPIC$CHR), s1=fData_EPIC$MAPINFO, s2=fData_EPIC$MAPINFO)
dim(pos_dat <- pos_dat[match(rownames(betas), pos_dat$geneid),]) #803777      5
fwrite(pos_dat, "~/PD_GRS/mQTL/probe_pos.txt", sep = " ", quote = F)

# methy
#betas_write <- betas %>% as.data.frame() %>% rownames_to_column("cpg")

fwrite(betas %>% as.data.frame() %>% rownames_to_column("cpg"), "~/PD_GRS/mQTL/methylation.txt", sep = " ", quote = F)
fwrite(betas[, terre_meta$men == 1] %>% as.data.frame() %>% rownames_to_column("cpg"), "~/PD_GRS/mQTL/male_methylation.txt", sep = " ", quote = F)
fwrite(betas[, terre_meta$men == 0] %>% as.data.frame() %>% rownames_to_column("cpg"), "~/PD_GRS/mQTL/female_methylation.txt", sep = " ", quote = F)

# SNP POS
write_delim(geno_df[geno_df$CHR == "chr21", c("SNP", "CHR", "POS")], "~/PD_GRS/mQTL/snp_pos_chr21.txt")

# SNPs
geno_vars <- colnames(geno_df)[-c(1, 3:6)]
write_delim(geno_df[geno_df$chr=="CHR21", geno_vars], "~/PD_GRS/mQTL/all_imputed_matrixeQTL_chr21.txt")

# Methy POS
write_delim(pos_dat[pos_dat$chr == "chr21",], "~/PD_GRS/mQTL/probe_pos_chr21.txt")
length(chr21_cpg <- pos_dat[pos_dat$chr == "chr21","Name"]) #9630

# methy
write_delim(betas %>% as.data.frame() %>% rownames_to_column("cpg") %>% filter(cpg %in% chr21_cpg), "~/PD_GRS/mQTL/methylation_chr21.txt")
```

All inputs for MatrixEQTL are now generated. Refer to scripts "run_matrixEQTL.R" and "run_mQTL_terre.sh" to generate results prior to the next section.

# cis-mQTL in TERRE

## CpGs from GRS EWAS with an mQTL
```{r pc check, eval=F}
mqtl <- fread("~/PD_GRS/mQTL/cis_mQTL_out.txt.gz", tmpdir="~/tmp") #mQTL detected at p < 0.25
length(unique(mqtl$SNP))
#[1] 7,786,044
length(unique(mqtl$gene))
#800,029
# most of the SNPs and CpGs are reported to have an mQTL relationship at p < 0.25 within 75kb

colnames(mqtl)[c(2,6)] <- c("TargetID","FDR.mQTL")
mqtl$Bonferroni.mQTL <- p.adjust(mqtl$`p-value`, method="bonferroni", n=361185165)
#361,185,165 tests (from ntest file output by matrixeQTL)

grs_res <- read.csv("~/PD_GRS/EWAS_TERRE/EWAS_GRS_cross_sex_hits.csv")
grs_mqtl <- left_join(grs_res, mqtl[,c(1,2,6,7)], by="TargetID")
nrow(grs_mqtl <- grs_mqtl[grs_mqtl$FDR.mQTL<=0.05,]) #13,342
#nrow(grs_mqtl <- grs_mqtl[grs_mqtl$Bonferroni.mQTL<=0.05,]) #12,250

cpg_mqtl <- summary(as.factor(grs_mqtl$TargetID))
#cg00891649 cg00916973 cg01341218 cg02478991 cg04226788 cg05314706 cg06721910 cg07368061 cg09793084 cg09860564 
#       758        198        292        700        740        295        778        705        656        787 
#cg12609785 cg15072451 cg15633388 cg16228356 cg17117718 cg17309384 cg17911788 cg18228076 cg18815117 cg18878992 
#       297        768        732        650        298         19        628        786        472        769 
#cg19832721 cg20163478 cg23519755 cg24319718 cg27060340 
#       721        299        708          1        285 

cpg_plot <- data.frame(Probe=names(cpg_mqtl), Num_mQTL=cpg_mqtl)
cpg_plot <- cpg_plot %>% arrange(desc(Num_mQTL))
cpg_plot$Probe <- factor(cpg_plot$Probe,levels=cpg_plot$Probe)
ggplot(cpg_plot, aes(x=Probe,y=Num_mQTL)) + geom_bar(stat="identity") + ylab("Number of associated mQTLs") + theme_classic() + theme(axis.text.x=element_text(angle=90), axis.title.x=element_blank()) + ggtitle("TERRE GRS-associated CpGs with mQTLs")

#what are the locations of the mQTL?
snp_pos <- fread("~/PD_GRS/mQTL/snp_pos.txt", tmpdir="~/tmp")
colnames(snp_pos)[2:3] <- c("SNP.CHR","SNP.POS")
grs_mqtl <- left_join(grs_mqtl, snp_pos, by="SNP")
summary(as.factor(grs_mqtl$SNP.CHR))
#chr10 chr17  chr2 
#   19 13322     1 

#how far are the mQTL on average from the CpG?
grs_mqtl$CHR <- paste("chr", grs_mqtl$CHR, sep="")
all(grs_mqtl$SNP.CHR==grs_mqtl$CHR) #TRUE
grs_mqtl$dist_mQTL <- grs_mqtl$SNP.POS - grs_mqtl$MAPINFO
summary(grs_mqtl$dist_mQTL/1000) #distance in kb
#    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
#-74.9950 -36.2105  -0.5325  -0.1056  36.0875  74.9930 

summary(abs(grs_mqtl$dist_mQTL/1000))
#   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
#   0.00   16.42   36.12   36.16   55.02   75.00 
   
#how many of the GRS SNPs are mQTL?
risk_scores <- read.delim("~/SHARE_DECIPHER/pd_grs_ewas_integration/prs_analyses/pd_prs_file.txt", header=FALSE, sep=" ") #107 SNPs (why not 90?)
#write snps to a txt file for LD calculation
write.table(risk_scores$V1, file="~/PD_GRS/mQTL/prs_snps.txt", row.names=FALSE, quote=FALSE, col.names=FALSE)

nrow(risk_scores[risk_scores$V1 %in% grs_mqtl$SNP,]) #4 SNPs are mQTL for the top CpGs
grs_mqtl$mQTL_PD_SNP <- (grs_mqtl$SNP %in% risk_scores$V1)
summary(grs_mqtl$mQTL_PD_SNP) #13 of the mQTL are SNPs
#   Mode   FALSE    TRUE 
#logical   13329      13 
length(unique(grs_mqtl[grs_mqtl$mQTL_PD_SNP==TRUE,"SNP"]))
#[1] 4

length(unique(grs_mqtl$TargetID)) #25
length(unique(grs_mqtl$SNP)) #3890

write.csv(grs_mqtl, file="~/PD_GRS/mQTL/GRS_EWAS_hits_mQTL.csv", row.names=F)
write.csv(grs_mqtl[grs_mqtl$mQTL_PD_SNP==TRUE,c("SNP","FDR.mQTL","Bonferroni.mQTL","TargetID","delta_M","adj.P.Val")], file="~/PD_GRS/mQTL/GRS_mQTL_TERRE.csv", row.names=F)

## Jan 20, 2025
#how many of the SNPs in LD to the GRS SNPs are mQTL?
grs_ld <- read.table("~/PD_GRS/mQTL/grs_snps_ld.vcor")
#add column names: "A" for the first SNP, B for the second SNP (in each pair).
colnames(grs_ld) <- c("CHR_A", "BP_A", "SNP_A", "CHR_B", "BP_B", "SNP_B", "R2")
#sanity check SNPs
length(unique(grs_ld$SNP_A)) #100
all(grs_ld$SNP_A %in% risk_scores$V1) #TRUE
length(unique(grs_ld$SNP_B)) #25,669
summary(grs_ld$R2)
#   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
# 0.2000  0.2681  0.4426  0.4934  0.6679  1.0000 

grs_mqtl <- read.csv("~/PD_GRS/mQTL/GRS_EWAS_hits_mQTL.csv")
nrow(grs_ld[grs_ld$SNP_B %in% grs_mqtl$SNP,]) #19,429 SNPs in LD (>0.2) to GRS are mQTL for the top CpGs
nrow(grs_ld[grs_ld$SNP_B %in% grs_mqtl$SNP & grs_ld$R2>0.5,]) #9,883 SNPs in LD (>0.5) to GRS are mQTL for the top CpGs
nrow(grs_ld[grs_ld$SNP_B %in% grs_mqtl$SNP & grs_ld$R2>0.8,]) #3,230 SNPs in LD (>0.8) to GRS are mQTL for the top CpGs

grs_mqtl$mQTL_GRS_LD <- (grs_mqtl$SNP %in% grs_ld$SNP_B)
summary(grs_mqtl$mQTL_GRS_LD)
#   Mode   FALSE    TRUE 
#logical     714   12628 
length(unique(grs_mqtl[grs_mqtl$mQTL_GRS_LD==TRUE,"SNP"]))
#3,584 mQTL are SNPs in LD to the GRS that associate with DNAm

length(unique(grs_mqtl$TargetID)) #25
length(unique(grs_mqtl$SNP)) #3890

write.csv(grs_mqtl, file="~/PD_GRS/mQTL/GRS_EWAS_hits_mQTL_LD.csv", row.names=F)
write.csv(grs_mqtl[grs_mqtl$mQTL_GRS_LD==TRUE,c("SNP","FDR.mQTL","Bonferroni.mQTL","TargetID","delta_M","adj.P.Val")], file="~/PD_GRS/mQTL/GRS_LD_mQTL_TERRE.csv", row.names=F)
```
![Number of associated mQTLs for TERRE GRS CpGs](/home1/NEURO/schaffner/PD_GRS/mQTL/grs_cpgs_mqtls.png)

## Colocalization analysis
Dec 30, 2024

For the 4 GRS SNPs which are also mQTL in TERRE, perform colocalization analysis to assess the strength of evidence for these SNPs being true mQTLs, and to determine whether the associations of the SNPs with PD and with DNAm are independent or not.

The script ~/PD_GRS/coloc/run_colocalization_SS.R was adapted from the previous version developed by Will Casazza, with some minor formatting changes. Arguments needed to run the script and save the output are included. The main function, coloc.abf, returns data frames called "summary" (saved to _colocalization_ph4.txt.gz) and "results" (saved to _colocalization_per_snp.txt.gz).

From "coloc.abf" documentation:

"summary is a vector giving the number of SNPs analysed, and **the posterior probabilities of H0 (no causal variant), H1 (causal variant for trait 1 only), H2 (causal variant for trait 2 only), H3 (two distinct causal variants) and H4 (one common causal variant)**"

"results is an annotated version of the input data containing log Approximate Bayes Factors and intermediate calculations, and the posterior probability SNP.PP.H4 of the SNP being causal for the shared signal if H4 is true. This is only relevant if the posterior support for H4 in summary is convincing."

```{r coloc, eval=F}
#source("~/PD_GRS/coloc/run_colocalization_SS.R")

#first, check posterior probabilities
coloc_ph4 <- fread("~/PD_GRS/coloc/cross_mqtl_cross_sumstats_pd_snp_colocalization_ph4.txt.gz")
head(coloc_ph4)
#   nsnps    PP.H0.abf    PP.H1.abf    PP.H2.abf   PP.H3.abf    PP.H4.abf      probe locus locus_snp
#1:   228 4.718758e-29 9.904247e-01 2.415786e-31 0.005066008 4.509299e-03 cg00340179     1  rs356203
#2:   392 2.450009e-04 9.872223e-01 7.942090e-07 0.003190896 9.341020e-03 cg00379420     1  rs356203
#3:   196 1.879186e-43 5.492019e-35 3.421668e-09 0.999999989 7.337095e-09 cg00538060     1  rs356203
#4:   136 1.555653e-02 9.699517e-01 3.917218e-05 0.002430368 1.202221e-02 cg00907891     1  rs356203
#5:   527 9.896689e-46 1.469541e-22 6.734545e-24 1.000000000 8.221536e-13 cg01966878     1  rs356203
#6:   198 5.955894e-09 5.449637e-07 1.053193e-02 0.963644303 2.582322e-02 cg02262221     1  rs356203

#PP.H4.abf is the posterior probability of H4 being true (one common causal variant)
#WC considered pp > 90% to be strong evidence
mqtl_snp <- unique(grs_mqtl[grs_mqtl$mQTL_PD_SNP==TRUE,"SNP"]) #get the 4 GRS SNPs that are mQTL in TERRE
nrow(coloc_sub <- coloc_ph4[coloc_ph4$PP.H4.abf>0.9 & coloc_ph4$locus_snp %in% mqtl_snp,])  #14
length(unique(coloc_sub$locus)) #1 locus has evidence for a SNP being a shared causal variant for PD risk (in Nalls dataset) and TERRE mQTL
length(unique(coloc_sub$probe)) #this SNP is an mQTL for 14 CpGs

length(unique(grs_mqtl$TargetID)) #25
nrow(coloc_sub[coloc_sub$probe %in% grs_mqtl$TargetID,]) #the GRS SNP is an mQTL for 6 of the CpGs that came up in the cross-sex GRS EWAS
#   nsnps    PP.H0.abf    PP.H1.abf    PP.H2.abf   PP.H3.abf PP.H4.abf      probe locus locus_snp
#1:   114 2.303989e-43 2.301545e-15 2.287531e-31 0.001286391 0.9987136 cg00916973    71 rs7225002
#2:    90 1.562966e-33 3.163633e-17 3.813858e-18 0.076273385 0.9237266 cg01341218    71 rs7225002
#3:    89 8.086731e-70 2.213930e-17 1.900894e-54 0.051092481 0.9489075 cg12609785    71 rs7225002
#4:   120 7.328324e-45 1.122008e-17 5.298514e-29 0.080203447 0.9197966 cg15633388    71 rs7225002
#5:    91 2.762007e-73 2.183197e-17 6.862954e-58 0.053300729 0.9466993 cg17117718    71 rs7225002
#6:    38 3.723654e-25 7.353284e-16 1.340865e-11 0.025504225 0.9744958 cg17911788    71 rs7225002

coloc_sub$cross_sex_ewas <- (coloc_sub$probe %in% grs_mqtl$TargetID)

#how many overlap with godmc?
godmc <- fread("~/PD_GRS/coloc/godmc_assoc_meta_all_30Dec24.csv.gz")
snp_pos <- fread("~/PD_GRS/mQTL/snp_pos.txt")
snp_pos$snp_id <- paste(snp_pos$CHR, snp_pos$POS, "SNP", sep=":")
colnames(coloc_sub)[9] <- "SNP"
library(dplyr)
coloc_sub <- left_join(coloc_sub, snp_pos[,c("SNP","snp_id")], by="SNP")
coloc_sub$godmc <- (coloc_sub$snp_id %in% godmc$snp)
summary(coloc_sub$godmc)
#   Mode    TRUE 
#logical      14 

godmc_sub <- godmc[godmc$snp %in% coloc_sub$snp_id,]
write.csv(godmc_sub, file="~/PD_GRS/coloc/godmc_SNPs_coloc.csv", row.names=F)
write.csv(coloc_sub, file="~/PD_GRS/coloc/PP.H4.abf_rs7225002.csv", row.names=F)
rm(godmc, snp_pos)

#check overlap with Ohlei et al.
ohlei <- read.csv("~/PD_GRS/Ohlei_2023_PD_SMR_blood.csv")
coloc_sub$ohlei_snp <- (coloc_sub$SNP %in% ohlei$topSNP)
coloc_sub$ohlei_cpg <- (coloc_sub$probe %in% ohlei$Probe)
summary(coloc_sub$ohlei_snp)
#   Mode   FALSE 
#logical      14 
summary(coloc_sub$ohlei_cpg)
#   Mode   FALSE    TRUE 
#logical       5       9 
write.csv(coloc_sub, file="~/PD_GRS/coloc/PP.H4.abf_rs7225002.csv", row.names=F)

#coloc_per_snp <- fread("~/PD_GRS/coloc/cross_mqtl_cross_sumstats_pd_snp_colocalization_per_snp.txt.gz")
```
