---
title: "TERRE GRS: Prediction of case-control status"
output: html_notebook
---

The manuscript "A Parkinson’s disease genetic risk score associates with blood DNAm on chromosome 17" explores how a genetic risk score (GRS) for Parkinson's disease (PD) associates with blood DNA methylation in the TERRE study of French agricultural workers, and evaluates GRS associations with DNAm in each sex as well as their sensitivity to non-genetic factors. This was originally spearheaded by Will Casazza and was posted as a pre-print on medRxiv in 2023. The current revision (2024, led by Sam Schaffner) aims to address previous reviewer feedback and prepare the manuscript for re-submission.


This script evaluates how well the GRS can predict case-control status, in a cross-sex sample and sex-stratified samples. The 2024 update includes adjustment to the models where the 10 DNAm PCs are replaced with batch (plate and row, chip excluded for association with sex) and robust PCs of DNAm-predicted cell type proportions. 


Last updated June 16, 2024 by Samantha Schaffner


Based on code by Will Casazza

```{r setup, include=FALSE, eval=F}
library(lumi)
library(limma)
library(minfi)
library(rcompanion)
library(readxl)
library(tidyverse)
library(ggpubr)
library(glue)
library(data.table)
library(car)
library(bacon)
library(ggplot2)
library(ggrepel)
library(dplyr)
library(IlluminaHumanMethylationEPICanno.ilm10b4.hg19)
source("~/kobor_shared_coding_resource/qqplot_NG.R")
knitr::opts_chunk$set(echo = TRUE)
```

# PD risk score data and matching

### Loading in and preparing data

This code reads in the Nalls 2019 PD risk score loci and organizes it into a table with alleles and weights for each SNP. Here, I matched the meta data with the CMR project meta data that includes cell type PCs and batch information.
```{r risk score dat, eval=FALSE}
risk_scores <- read.delim("~/SHARE_DECIPHER/pd_grs_ewas_integration/prs_analyses/pd_prs_file.txt", header=FALSE, sep=" ")
head(risk_scores)
#           V1 V2      V3
#1 rs114138760  C  0.2812
#2  rs35749011  A  0.6068
#3  rs76763715  T -0.7467
#4   rs6658353  C  0.0650
#5  rs11578699  T -0.0704
#6    rs823118  T  0.1066

terre_prs <- fread("~/SHARE_DECIPHER/pd_grs_ewas_integration/prs_analyses/terre_prs.sscore")
# WC: this is the GRS generated with pT = 5e-8

terre_pcs <- fread("~/SHARE_DECIPHER/pd_grs_ewas_integration/TERRE_QC/raw_data.geno.maf.mind.sex_check.het_filter.ibd_filter.eigenvec")
#Columns V1 and V2 are ID labels, V3-V22 are PCs
terre_metadata <- read.csv("/home1/NEURO/SHARE_DECIPHER/terre_meta_master.csv") %>%
  mutate(
    FID = gsub("(PAE_[0-9]*_[1-9]*)_.*", "\\1", FID),
    IID = gsub("(PAE_[0-9]*_[1-9]*)_.*", "\\1", IID)
  )

all_data <- terre_prs %>%
  left_join(terre_metadata, by = c("IID")) %>%
  left_join(terre_pcs, by = c("IID" = "V1"))

# get genotype and DNAm PCs from the following file:
prsice_cov <- fread("~/SHARE_DECIPHER/pd_grs_ewas_integration/prs_analyses/prsice_cov_and_status_mvalues.txt")
all(prsice_cov$patient %in% all_data$patient) #TRUE
colnames(all_data) <- gsub("men", "sex", colnames(all_data))
meta <- left_join(prsice_cov, all_data, by=c("patient","PD","IID","sex","age","V3","V4","V5"))
colnames(meta) #rename DNAm and geno PC columns for clarity
colnames(meta)[1:10] <- paste("DNAm", colnames(meta)[1:10], sep="_")
colnames(meta)[13:15] <- paste("geno", colnames(meta)[13:15], sep="_")
meta <- meta[,-23] #remove extra dummy column "X"
colnames(meta)

# add batch and cell type variables
propensity_meta_F <- read.csv("~/SHARE_DECIPHER/processed_DNAm_data/2022/TERRE_processed_2022/3-terre_meta_F_probit_matched_4.5.3.csv")
propensity_meta_M <- read.csv("~/SHARE_DECIPHER/processed_DNAm_data/2022/TERRE_processed_2022/3-terre_meta_M_probit_matched_4.5.3.csv")
# fix sex columns
propensity_meta_F$reportedSex <- "F"
propensity_meta_M$reportedSex <- "M"
propensity_meta <- rbind(propensity_meta_F, propensity_meta_M)

# re-code and join
colnames(propensity_meta)[7] <- "sex"
meta$sex <- gsub(0, "F", gsub(1, "M", meta$sex))
all(propensity_meta$patient %in% meta$patient) #TRUE
cols <- colnames(meta)[colnames(meta) %in% colnames(propensity_meta)]
meta <- left_join(meta, propensity_meta, by=cols)
colnames(meta)
write.csv(meta, file="~/PD_GRS/meta_geno_DNAm_PCs.csv", row.names=F) 

# DNAm data
load("~/SHARE_DECIPHER/processed_DNAm_data/2022/TERRE_processed_2022/1-TERRE_RG_filtered.RData")
mvals <- getM(PD_RG_filtered)
```

# Testing how well GRS explains case-control status

Prior to EWAS, testing GRS performance here in TERRE (cross-sex and sex-stratified). I updated the models to adjust for cell type and batch (as opposed to 10 DNAm PCs).

GRS performance is assessed here by comparing models predicting case-control status that include or do not include the GRS. The nagelkerke() function from the "rcompanion" package produces pseudo R-squared measurements, which indicate how well each model explains the data relative to each other. A higher R squared indicates better prediction of the outcome. The Nagelkerke measure is an more interpretable adjustment of the Cox and Snell measure, where the upper bound is 1.

Liability scale heritability for the PD GRS is also calculated, which assumes there is a continuous distribution of risk for PD that is influenced by genetic and environmental factors. Heritability is interpreted here as the proportion of variance in PD risk that is due to genetics.

## Old model with genotype PCs only

First, I will check if I get the same numbers Will previously generated, starting with a model that includes genotype PCs and not DNAm PCs (since there was a copy of this script annnotated as such).
```{r old model, eval=F}
fit1_old <- glm(PD ~ SCORE1_AVG + sex + age + geno_V3 + geno_V4 + geno_V5, data = meta, family = "binomial")
summary(fit1_old) #PRS: p = 0.002
fit1_null_old <- glm(PD ~ sex + age + geno_V3 + geno_V4 + geno_V5, data = meta, family = "binomial")
summary(fit1_null_old) #nothing sig
(cross_r2_old <- nagelkerke(fit1_old, null = fit1_null_old))
#                             Pseudo.R.squared
#McFadden                            0.0366160
#Cox and Snell (ML)                  0.0446199
#Nagelkerke (Cragg and Uhler)        0.0626223

# liability R2
lin1 <- lm(PD ~ SCORE1_AVG + sex + age + geno_V3 + geno_V4 + geno_V5, data = meta)
lin2 <- lm(PD ~ sex + age + geno_V3 + geno_V4 + geno_V5, data = meta)
R2 <- (deviance(lin2) - deviance(lin1)) / deviance(lin2)
h2 <- pd_h2(R2, sum(all_data$status) / nrow(all_data), 5 / 1000)
R2 #0.04451659
h2 #0.02456678
```
These values are very close to what was reported in the manuscript.

## Old model with genotype and DNAm PCs
```{r DNAm PC model, eval=F}
fit1_old2 <- glm(PD ~ SCORE1_AVG + sex + age + geno_V3 + geno_V4 + geno_V5 + DNAm_PC1 + DNAm_PC2 + DNAm_PC3 + DNAm_PC4 + DNAm_PC5 + DNAm_PC6 + DNAm_PC7 + DNAm_PC8 + DNAm_PC9 + DNAm_PC10, data = meta, family = "binomial")
summary(fit1_old2) #PRS: p = 0.002
fit1_null_old2 <- glm(PD ~ sex + age + geno_V3 + geno_V4 + geno_V5 + DNAm_PC1 + DNAm_PC2 + DNAm_PC3 + DNAm_PC4 + DNAm_PC5 + DNAm_PC6 + DNAm_PC7 + DNAm_PC8 + DNAm_PC9 + DNAm_PC10, data = meta, family = "binomial")
summary(fit1_null_old2) #nothing sig
(cross_r2_old2 <- nagelkerke(fit1_old2, null = fit1_null_old2))
#                             Pseudo.R.squared
#McFadden                            0.0377788
#Cox and Snell (ML)                  0.0448697
#Nagelkerke (Cragg and Uhler)        0.0637953

# liability R2
lin1 <- lm(PD ~ SCORE1_AVG + sex + age + geno_V3 + geno_V4 + geno_V5  + DNAm_PC1 + DNAm_PC2 + DNAm_PC3 + DNAm_PC4 + DNAm_PC5 + DNAm_PC6 + DNAm_PC7 + DNAm_PC8 + DNAm_PC9 + DNAm_PC10, data = meta)
lin2 <- lm(PD ~ sex + age + geno_V3 + geno_V4 + geno_V5 + DNAm_PC1 + DNAm_PC2 + DNAm_PC3 + DNAm_PC4 + DNAm_PC5 + DNAm_PC6 + DNAm_PC7 + DNAm_PC8 + DNAm_PC9 + DNAm_PC10, data = meta)
R2 <- (deviance(lin2) - deviance(lin1)) / deviance(lin2)
h2 <- pd_h2(R2, sum(all_data$status) / nrow(all_data), 5 / 1000)
R2 #0.04566368
h2 #0.02522456
```
These numbers are a little closer to what was reported in the paper, but still not exactly the same.

## New model with batch and cell type
```{r GRS case-control new, eval=F}
#heritability calculation
pd_h2 <- function(R2O, case_prop, prevalence = 0.22) {
  K <- prevalence
  P <- case_prop
  thd <- -1 * qnorm(K, 0, 1)
  zv <- dnorm(thd) # z (normal density)
  mv <- zv / K # mean liability for case
  theta <- mv * (P - K) / (1 - K) * (mv * (P - K) / (1 - K) - thd) # θ in equation
  cv <- K * (1 - K) / zv^2 * K * (1 - K) / (P * (1 - P)) # C in
  return(R2O * cv / (1 + R2O * theta * cv))
}

# R2
#prev: binomial regression for PD ~ PRS + sex + age + geno PCs
#in manuscript it also includes 10 DNAm PCs, which wasn't coded here
# new: PD ~ PRS + sex + age + geno PCs + cell type PCs 1-6 + plate + row
fit1 <- glm(PD ~ SCORE1_AVG + sex + age + geno_V3 + geno_V4 + geno_V5 + CTP_PC1 + CTP_PC2 + CTP_PC3 + CTP_PC4 + CTP_PC5 + CTP_PC6 + plate + SentrixPosition_A, data = meta, family = "binomial")
summary(fit1) #PRS: p = 0.005; CTP PC2: p = 0.01
fit1_null <- glm(PD ~ sex + age + geno_V3 + geno_V4 + geno_V5 + CTP_PC1 + CTP_PC2 + CTP_PC3 + CTP_PC4 + CTP_PC5 + CTP_PC6 + plate + SentrixPosition_A, data = meta, family = "binomial")
summary(fit1_null) #CTP PC2: p = 0.004
(cross_r2 <- nagelkerke(fit1, null = fit1_null))
#                             Pseudo.R.squared
#McFadden                            0.0329916
#Cox and Snell (ML)                  0.0377029
#Nagelkerke (Cragg and Uhler)        0.0547970

#"pseudo R-squared measures are relative measures among similar models indicating how well the model explains the data"
# it is only relevant relative to another model; a higher R2 indicates the model better predicts the outcome
# in this case, the real model was 0.03-0.05 higher relative to the null
#Nagelkerke (Cragg and Uhler) value used for plots and reporting

# liability R2 for updated cross-sex model
lin1 <- lm(PD ~ SCORE1_AVG + sex + age + geno_V3 + geno_V4 + geno_V5 + CTP_PC1 + CTP_PC2 + CTP_PC3 + CTP_PC4 + CTP_PC5 + CTP_PC6 + plate + SentrixPosition_A, data = meta)
lin2 <- lm(PD ~ sex + age + geno_V3 + geno_V4 + geno_V5 + CTP_PC1 + CTP_PC2 + CTP_PC3 + CTP_PC4 + CTP_PC5 + CTP_PC6 + plate + SentrixPosition_A, data = meta)
R2 <- (deviance(lin2) - deviance(lin1)) / deviance(lin2)
h2 <- pd_h2(R2, sum(all_data$status) / nrow(all_data), 5 / 1000)
R2 #0.03863557
h2 #0.02121458
```

## New models, sex-stratified
```{r sex strat r2, eval=F}
fit1_male <- glm(PD ~ SCORE1_AVG + age + geno_V3 + geno_V4 + geno_V5 + CTP_PC1 + CTP_PC2 + CTP_PC3 + CTP_PC4 + CTP_PC5 + CTP_PC6 + plate + SentrixPosition_A, data = meta[sex == "M", ], family = "binomial")
summary(fit1_male) #CTP PC2: p = 0.01
fit1_male_null <- glm(PD ~ age + geno_V3 + geno_V4 + geno_V5 + CTP_PC1 + CTP_PC2 + CTP_PC3 + CTP_PC4 + CTP_PC5 + CTP_PC6 + plate + SentrixPosition_A, data = meta[sex == "M", ], family = "binomial")
summary(fit1_male_null) #CTP PC2: p = 0.01
(male_r2 <- nagelkerke(fit1_male, null = fit1_male_null))
#                             Pseudo.R.squared
#McFadden                            0.0200165
#Cox and Snell (ML)                  0.0201489
#Nagelkerke (Cragg and Uhler)        0.0315673

# liability R2 for updated male model
lin1 <- lm(PD ~ SCORE1_AVG + age + geno_V3 + geno_V4 + geno_V5 + CTP_PC1 + CTP_PC2 + CTP_PC3 + CTP_PC4 + CTP_PC5 + CTP_PC6 + plate + SentrixPosition_A, data = meta[sex == "M", ])
lin2 <- lm(PD ~ age + geno_V3 + geno_V4 + geno_V5 + CTP_PC1 + CTP_PC2 + CTP_PC3 + CTP_PC4 + CTP_PC5 + CTP_PC6 + plate + SentrixPosition_A, data = meta[sex == "M", ])
R2 <- (deviance(lin2) - deviance(lin1)) / deviance(lin2)
h2 <- pd_h2(R2, sum(all_data$status) / nrow(all_data), 5 / 1000)
R2 #0.02566047
h2 #0.01393612

fit1_female <- glm(PD ~ SCORE1_AVG + age + geno_V3 + geno_V4 + geno_V5 + CTP_PC1 + CTP_PC2 + CTP_PC3 + CTP_PC4 + CTP_PC5 + CTP_PC6 + plate + SentrixPosition_A, data = meta[sex == "F", ], family = "binomial")
summary(fit1_female) #PRS p = 0.02
fit1_female_null <- glm(PD ~ age + geno_V3 + geno_V4 + geno_V5 + CTP_PC1 + CTP_PC2 + CTP_PC3 + CTP_PC4 + CTP_PC5 + CTP_PC6 + plate + SentrixPosition_A, data = meta[sex == "F", ], family = "binomial")
summary(fit1_female_null) #nothing sig
(female_r2 <- nagelkerke(fit1_female, null = fit1_female_null))
#                             Pseudo.R.squared
#McFadden                            0.0530105
#Cox and Snell (ML)                  0.0578192
#Nagelkerke (Cragg and Uhler)        0.0856752

# liability R2 for updated female model
lin1 <- lm(PD ~ SCORE1_AVG + age + geno_V3 + geno_V4 + geno_V5 + CTP_PC1 + CTP_PC2 + CTP_PC3 + CTP_PC4 + CTP_PC5 + CTP_PC6 + plate + SentrixPosition_A, data = meta[sex == "F", ])
lin2 <- lm(PD ~ age + geno_V3 + geno_V4 + geno_V5 + CTP_PC1 + CTP_PC2 + CTP_PC3 + CTP_PC4 + CTP_PC5 + CTP_PC6 + plate + SentrixPosition_A, data = meta[sex == "F", ])
R2 <- (deviance(lin2) - deviance(lin1)) / deviance(lin2)
h2 <- pd_h2(R2, sum(all_data$status) / nrow(all_data), 5 / 1000)
R2 #0.0604893
h2 #0.03384384
```

### Plotting cross-sex and sex-stratified GRS models
```{r plot GRS, eval=F}
r2_labels <- c(
  0.0548,
  round(male_r2$Pseudo.R.squared.for.model.vs.null[3], 4),
  round(female_r2$Pseudo.R.squared.for.model.vs.null[3], 4)
)
h2_labels <- c(
  0.0212,
  round(pd_h2(male_r2$Pseudo.R.squared.for.model.vs.null[3], sum(all_data$status) / nrow(all_data), 5 / 1000), 4),
  round(pd_h2(female_r2$Pseudo.R.squared.for.model.vs.null[3], sum(all_data$status) / nrow(all_data), 5 / 1000), 4)
)

to_plot <- meta %>%
  mutate(
    PD = ifelse(PD, "CASE", "CONTROL"),
    Sex = ifelse(sex == "M", "Male", "Female")
  ) %>%  mutate(Sex = factor(Sex, levels = c("Cross-sex", "Male", "Female")))
cross_data <- meta %>% mutate(PD = ifelse(PD, "CASE", "CONTROL"))
cross_data$Sex <- "Cross-sex"
to_plot <- rbind(cross_data, to_plot)

png("~/PD_GRS/prs_plots_by_PD_sex.png", width=600, height=500)
ggboxplot(
  to_plot,
  x = "PD",
  y = "SCORE1_AVG",
  color = "Sex",
  add = "jitter"
) +
  facet_grid(~Sex, labeller = label_bquote(cols = .(as.character(Sex)) ~ R^2 ~ ": " ~ .(r2_labels[Sex]) ~ ";" ~ h^2 ~ ": " ~ .(h2_labels[Sex]))) +
  labs(y = "Polygenic Risk Score", x = "Parkinson's Diagnosis") +
  scale_color_manual(values = c("gray", "light blue", "pink"))
dev.off()

png("~/PD_GRS/prs_plots_by_sex.png", width=300, height=500)
ggboxplot(
  to_plot,
  x = "Sex",
  y = "SCORE1_AVG",
  color = "Sex",
  add = "jitter"
) +
  stat_compare_means() +

  labs(y = "Polygenic Risk Score") +
  scale_color_manual(values = c("Male" = "light blue", "Female" = "pink"))
dev.off()
```
![PRS by PD status and sex](/home1/NEURO/schaffner/PD_GRS/prs_plots_by_PD_sex.png)

![PRS by sex (cases and controls combined)](/home1/NEURO/schaffner/PD_GRS/prs_plots_by_sex.png)

## Additional sensitivity tests 
Interactions with PRS x sex, PRS x smoking, PRS x pesticides, and PRS x cell type PC2 will be explored here to see if they better explain PD status.
```{r GRS sensitivity, eval=F}
fit2 <- glm(PD ~ SCORE1_AVG * sex + age + geno_V3 + geno_V4 + geno_V5 + CTP_PC1 + CTP_PC2 + CTP_PC3 + CTP_PC4 + CTP_PC5 + CTP_PC6 + plate + SentrixPosition_A, data = meta, family = "binomial")
summary(fit2) #PRS p = 0.02, CTP PC2 p = 0.01, interaction p = 0.51
fit3 <- glm(PD ~ SCORE1_AVG * smoking + sex + age + geno_V3 + geno_V4 + geno_V5 + CTP_PC1 + CTP_PC2 + CTP_PC3 + CTP_PC4 + CTP_PC5 + CTP_PC6 + plate + SentrixPosition_A, data = meta, family = "binomial")
summary(fit3) # PRS p = 0.25, smoking p = 0.01, CTP PC2 p = 0.01, interaction p = 0.89
fit4 <- glm(PD ~ SCORE1_AVG * pesticides + sex + age + geno_V3 + geno_V4 + geno_V5 + CTP_PC1 + CTP_PC2 + CTP_PC3 + CTP_PC4 + CTP_PC5 + CTP_PC6 + plate + SentrixPosition_A, data = meta, family = "binomial")
summary(fit4) #PRS p = 0.16, pest p = 0.51, geno V4 p = 0.05, interaction p = 0.73

#PRS x cell type PC2, sex x cell type PC2
fit5 <- glm(PD ~ SCORE1_AVG * CTP_PC2 + sex + age + geno_V3 + geno_V4 + geno_V5 + CTP_PC1 + CTP_PC3 + CTP_PC4 + CTP_PC5 + CTP_PC6 + plate + SentrixPosition_A, data = meta, family = "binomial")
summary(fit5) #PRS p = 0.002, CTP PC2 p = 0.038, interaction p = 0.15
fit6 <- glm(PD ~ SCORE1_AVG + age + geno_V3 + geno_V4 + geno_V5 + CTP_PC1 + sex*CTP_PC2 + CTP_PC3 + CTP_PC4 + CTP_PC5 + CTP_PC6 + plate + SentrixPosition_A, data = meta, family = "binomial") #PRS p = 0.004, interaction p = 0.14
summary(fit6)

fit7 <- glm(PD ~ SCORE1_AVG * CTP_PC2 + age + geno_V3 + geno_V4 + geno_V5 + CTP_PC1 + CTP_PC3 + CTP_PC4 + CTP_PC5 + CTP_PC6 + plate + SentrixPosition_A, data = meta[sex=="M",], family = "binomial")
summary(fit7) #PRS p = 0.12, CTP PC2 p = 0.032, interaction p = 0.58

fit8 <- glm(PD ~ SCORE1_AVG * CTP_PC2 + age + geno_V3 + geno_V4 + geno_V5 + CTP_PC1 + CTP_PC3 + CTP_PC4 + CTP_PC5 + CTP_PC6 + plate + SentrixPosition_A, data = meta[sex=="F",], family = "binomial")
summary(fit8) #PRS p = 0.012, CTP PC2 p = 0.84, interaction p = 0.19

fit_null <- glm(PD ~ SCORE1_AVG, data = meta, family = "binomial")
summary(fit_null) #p = 0.0039

png("~/PD_GRS/CTP_PC2_plots_by_PD_sex.png", width=600, height=500)
ggboxplot(
  to_plot,
  x = "PD",
  y = "CTP_PC2",
  color = "Sex",
  add = "jitter"
) +
  facet_grid(~Sex) +
  labs(y = "Cell type PC2 loadings", x = "Parkinson's Diagnosis") +
  scale_color_manual(values = c("gray", "light blue", "pink"))
dev.off()
```
![Cell type PC 2 by PD status and sex](/home1/NEURO/schaffner/PD_GRS/CTP_PC2_plots_by_PD_sex.png)

Cell type PC2 loadings differ by case-control status for males and the cross-sex sample.
