---
title: "TERRE GRS: Epigenome-wide asssociation analysis"
output: html_notebook
---

The manuscript "A Parkinsonâ€™s disease genetic risk score associates with blood DNAm on chromosome 17" explores how a genetic risk score (GRS) for Parkinson's disease (PD) associates with blood DNA methylation in the TERRE study of French agricultural workers, and evaluates GRS associations with DNAm in each sex as well as their sensitivity to non-genetic factors. This was originally spearheaded by Will Casazza and was posted as a pre-print on medRxiv in 2023. The current revision (2024, led by Sam Schaffner) aims to address previous reviewer feedback and prepare the manuscript for re-submission.


In this script, EWAS is conducted to identify DNAm patterns associated with the GRS in the cross-sex and sex-stratified samples in TERRE. The models are adjusted for batch, cell type, and disease status. Results will be compared with the previous models (adjusted for 10 DNAm PCs).


Last updated Aug 24, 2024 by Samantha Schaffner: Generating figures and converting delta M to delta beta

Based on code by Will Casazza

```{r setup, include=FALSE, eval=F}
library(lumi)
library(limma)
library(minfi)
library(rcompanion)
library(readxl)
library(tidyverse)
library(ggpubr)
library(glue)
library(data.table)
library(car)
library(bacon)
library(ggplot2)
library(ggrepel)
library(dplyr)
library(rtracklayer)
library(DescTools)
library(Gviz)
library(plyr)
library(IlluminaHumanMethylationEPICanno.ilm10b4.hg19)
source("~/kobor_shared_coding_resource/qqplot_NG.R")
knitr::opts_chunk$set(echo = TRUE)
```


### Loading in and preparing data
```{r load dat, eval=FALSE}
meta <- read.csv("~/PD_GRS/meta_geno_DNAm_PCs.csv")

# DNAm data (2022, most recent QC'ed, same as npj PD 2024 article)
load("~/SHARE_DECIPHER/processed_DNAm_data/2022/TERRE_processed_2022/1-TERRE_RG_filtered.RData")
mvals <- getM(PD_RG_filtered)
```

# Association analysis

## EWAS model checks

Prior to looking at results, I'll assess whether the new EWAS model seems appropriate using three measures:

*1. Variance inflation factor*: The variance inflation factor (VIF) can be used to detect collinearity. Higher VIF values indicate a greater degree of collinearity. Typically VIF < 5 is considered acceptable.


*2. Q-Q plot of nominal p-values*: If the variable of interest does not affect global DNAm, p-values should distribute close to the center line on a normal Q-Q plot, with some deviation at the tail indicating associations detected. The shape of the Q-Q plot can visually indicate model inflation or deflation.


*3. Bias and inflation calculations*: The "bacon" package (van Iterson et al., 2017) was developed to calculate and optionally, correct for, bias and inflation observed in epigenome-wide association studies. It models test statistics using a Bayesian mixture model, and estimates the model parameters: bias (mean) and inflation (standard deviation). The Bayesian mixture model is proposed to more accurately model the quantitative nature and technical/biological influences on epigenomic and transcriptomic data, as opposed to the traditional genomic inflation factor (based on discrete genotypes).


```{r ewas models, eval=F}
## Preparing data
# match meta data and M-values
meta <- meta[complete.cases(meta$CTP_PC1),]
mvals <- mvals[,match(meta$patient, colnames(mvals))]
all(colnames(mvals)==meta$patient) #TRUE

# subset meta data to relevant columns and re-format
meta_sub <- meta[,c("patient","SCORE1_AVG","PD","age","sex","geno_V3","geno_V4","geno_V5","CTP_PC1","CTP_PC2","CTP_PC3","CTP_PC4","CTP_PC5","CTP_PC6","smoking","pesticides","head_trauma","alcohol1","levodopa","agonist","plate","SentrixPosition_A")]
str(meta_sub)
meta_sub$PD <- as.factor(meta_sub$PD)
meta_sub$sex <- as.factor(meta_sub$sex)
meta_sub$smoking <- as.factor(meta_sub$smoking)
meta_sub$pesticides <- as.factor(meta_sub$pesticides)
meta_sub$head_trauma <- as.factor(meta_sub$head_trauma)
meta_sub$alcohol1 <- as.factor(meta_sub$alcohol1)
meta_sub$levodopa <- as.factor(meta_sub$levodopa)
meta_sub$plate <- as.factor(meta_sub$plate)
meta_sub$SentrixPosition_A <- as.factor(meta_sub$SentrixPosition_A)

## Variance inflation factor
# check for one CpG (result is the same as when applied to all CpGs and averaged)
vif(lm(mvals[1,] ~ meta_sub$SCORE1_AVG + meta_sub$age + meta_sub$sex + meta_sub$geno_V3 + meta_sub$geno_V4 + meta_sub$geno_V5 + meta_sub$CTP_PC1 + meta_sub$CTP_PC2 + meta_sub$CTP_PC3 + meta_sub$CTP_PC4 + meta_sub$CTP_PC5 + meta_sub$CTP_PC6 + meta_sub$plate + meta_sub$SentrixPosition_A + meta_sub$PD))[,3]

#       meta_sub$SCORE1_AVG               meta_sub$age               meta_sub$sex 
#                  1.047923                   1.104620                   1.541255 
#          meta_sub$geno_V3           meta_sub$geno_V4           meta_sub$geno_V5 
#                  1.081889                   1.103137                   1.068439 
#          meta_sub$CTP_PC1           meta_sub$CTP_PC2           meta_sub$CTP_PC3 
#                  1.096261                   1.133180                   1.316482 
#          meta_sub$CTP_PC4           meta_sub$CTP_PC5           meta_sub$CTP_PC6 
#                  1.142075                   1.244627                   1.160119 
#            meta_sub$plate meta_sub$SentrixPosition_A                meta_sub$PD 
#                  1.135407                   1.051032                   1.072224 


## Q-Q plot
pvals <- sapply(1:nrow(mvals), function(x) {
  coef(summary(lm(mvals[x,] ~ meta_sub$SCORE1_AVG + meta_sub$age + meta_sub$sex + meta_sub$geno_V3 + meta_sub$geno_V4 + meta_sub$geno_V5 + meta_sub$CTP_PC1 + meta_sub$CTP_PC2 + meta_sub$CTP_PC3 + meta_sub$CTP_PC4 + meta_sub$CTP_PC5 + meta_sub$CTP_PC6 + meta_sub$plate + meta_sub$SentrixPosition_A + meta_sub$PD)))[2,4]
   })
qqunif.plot(pvals, title="Model adjusted for PD, batch, and cell type")

## Bias and inflation (based on Z-values)
zvals <- qnorm(pvals)
bacon(zvals)
#...estimated bias: -0.088.
#...estimated inflation: 1.
```
![Q-Q plot for updated EWAS model](/home1/NEURO/schaffner/PD_GRS/EWAS_TERRE/qqplot.png)

The majority of p-values follow the expected distribution, while the tail branches off strongly. This is similar to the previous Q-Q plot generated by Will, and demonstrates an appropriate model (mostly null p-values) with an expected strong effect of genotype on DNA methylation.

Despite an association between cell type PC2 and PD status, the VIF calculations indicate low (if any) collinearity, and the bacon results indicate low bias (number should be near zero) and no inflation (number should be near 1, and is exactly 1 here).

## Differential methylation analysis

Will used limma on all EPIC probes passing QC, modeling the data with M-values. I'll continue with this approach and transform results back to beta values later for plotting and interpretation purposes.

First, as Will did, I'll apply the updated models to (1) the cross-sex sample, (2) males, (3) females, (4) PD cases, and (5) controls. This permits us to detect whether GRS associations with DNAm are dependent on sex and/or disease status.

```{r prs ewas, eval=F}
# set up meta data
na_row <- !apply(meta_sub[, c("SCORE1_AVG", "age", "sex", "geno_V3", "geno_V4", "geno_V5", "CTP_PC1", "CTP_PC2", "CTP_PC3", "CTP_PC4", "CTP_PC5", "CTP_PC6", "plate", "SentrixPosition_A", "PD")], 1, function(x) any(is.na(x)))
summary(na_row) #no NA's for these covariates
str(meta_sub[,c("SCORE1_AVG", "age", "sex", "geno_V3", "geno_V4", "geno_V5", "CTP_PC1", "CTP_PC2", "CTP_PC3", "CTP_PC4", "CTP_PC5", "CTP_PC6", "plate", "SentrixPosition_A", "PD")]) #checking format again - good

####### Cross-sex PRS EWAS
# limma design matrix and linear model fit
design_prs <- model.matrix(~ 1 + SCORE1_AVG + age + sex + geno_V3 + geno_V4 + geno_V5 + CTP_PC1 + CTP_PC2 + CTP_PC3 + CTP_PC4 + CTP_PC5 + CTP_PC6 + plate + SentrixPosition_A + PD, data = meta_sub)
prs_fit <- lmFit(mvals, design_prs)
prs_fit <- eBayes(prs_fit)
# transform adjusted M-values to adjusted beta-values (code from Kruppa et al. 2021)
prs_fit_coef <- prs_fit$coefficients
m0_eff <- prs_fit_coef[, "(Intercept)"] #M-value model intercept
m1_eff <- m0_eff + prs_fit_coef[, "SCORE1_AVG"] #M-value model intercept plus GRS coefficient (delta M)

prs_res <- limma::topTable(prs_fit, coef = 2, number = Inf, genelist=rownames(mvals))
prs_res$adjDB <- m2beta(m1_eff) - m2beta(m0_eff) #transform each adjusted M-value to a beta-value, and subtract intercept beta value from intercept + coefficient beta value to obtain the adjusted delta beta
nrow(prs_res[abs(prs_res$adjDB)>=0.03 & prs_res$adj.P.Val<=0.05,]) #23

#### check PRS model coefficients vs "logFC" column: is topTable outputting delta M?
coef <- prs_fit$coefficients
all(rownames(coef)%in%prs_res$ID) #TRUE
coef <- coef[match(prs_res$ID, rownames(coef)),]
all(rownames(coef)==prs_res$ID) #TRUE
all(coef[,2]==prs_res$logFC) #TRUE
#### logFC column is the model coefficient

####### Cases only
design_prs_case <- model.matrix(~ 1 + SCORE1_AVG + age + sex + geno_V3 + geno_V4 + geno_V5 + CTP_PC1 + CTP_PC2 + CTP_PC3 + CTP_PC4 + CTP_PC5 + CTP_PC6 + plate + SentrixPosition_A, data = meta_sub[meta_sub$PD=="1",])
prs_fit_case <- lmFit(mvals[,match(unlist(meta_sub[meta_sub$PD=="1","patient"]), colnames(mvals))], design_prs_case)
prs_fit_case <- eBayes(prs_fit_case)
# transform adjusted M-values to adjusted beta-values (code from Kruppa et al. 2021)
prs_fit_coef <- prs_fit_case$coefficients
m0_eff <- prs_fit_coef[, "(Intercept)"] #M-value model intercept
m1_eff <- m0_eff + prs_fit_coef[, "SCORE1_AVG"] #M-value model intercept plus GRS coefficient (delta M)

case_res <- limma::topTable(prs_fit_case, coef = 2, number = Inf, genelist=rownames(mvals))
case_res$adjDB <- m2beta(m1_eff) - m2beta(m0_eff) #transform each adjusted M-value to a beta-value, and subtract intercept beta value from intercept + coefficient beta value to obtain the adjusted delta beta
nrow(case_res[abs(case_res$adjDB)>=0.03 & case_res$adj.P.Val<=0.05,]) #0

# Controls only
design_prs_control <- model.matrix(~ 1 + SCORE1_AVG + age + sex + geno_V3 + geno_V4 + geno_V5 + CTP_PC1 + CTP_PC2 + CTP_PC3 + CTP_PC4 + CTP_PC5 + CTP_PC6 + plate + SentrixPosition_A, data = meta_sub[meta_sub$PD=="0",])
prs_fit_control <- lmFit(mvals[,match(unlist(meta_sub[meta_sub$PD=="0","patient"]), colnames(mvals))], design_prs_control)
prs_fit_control <- eBayes(prs_fit_control)
# transform adjusted M-values to adjusted beta-values (code from Kruppa et al. 2021)
prs_fit_coef <- prs_fit_control$coefficients
m0_eff <- prs_fit_coef[, "(Intercept)"] #M-value model intercept
m1_eff <- m0_eff + prs_fit_coef[, "SCORE1_AVG"] #M-value model intercept plus GRS coefficient (delta M)

control_res <- limma::topTable(prs_fit_control, coef = 2, number = Inf, genelist=rownames(mvals))
control_res$adjDB <- m2beta(m1_eff) - m2beta(m0_eff) #transform each adjusted M-value to a beta-value, and subtract intercept beta value from intercept + coefficient beta value to obtain the adjusted delta beta
nrow(control_res[abs(control_res$adjDB)>=0.03 & control_res$adj.P.Val<=0.05,]) #14

# Males
design_prs_male <- model.matrix(~ 1 + SCORE1_AVG + age + geno_V3 + geno_V4 + geno_V5 + CTP_PC1 + CTP_PC2 + CTP_PC3 + CTP_PC4 + CTP_PC5 + CTP_PC6 + plate + SentrixPosition_A + PD, data = meta_sub[meta_sub$sex=="M",])
prs_fit_male <- lmFit(mvals[,match(unlist(meta_sub[meta_sub$sex=="M","patient"]), colnames(mvals))], design_prs_male)
# Coefficients not estimable: plate6 plate9 
# Partial NA coefficients for 803777 probe(s)
prs_fit_male <- eBayes(prs_fit_male)
# Estimation of var.prior failed - set to default value
# transform adjusted M-values to adjusted beta-values (code from Kruppa et al. 2021)
prs_fit_coef <- prs_fit_male$coefficients
m0_eff <- prs_fit_coef[, "(Intercept)"] #M-value model intercept
m1_eff <- m0_eff + prs_fit_coef[, "SCORE1_AVG"] #M-value model intercept plus GRS coefficient (delta M)

male_res <- limma::topTable(prs_fit_male, coef = 2, number = Inf, genelist=rownames(mvals))
male_res$adjDB <- m2beta(m1_eff) - m2beta(m0_eff) #transform each adjusted M-value to a beta-value, and subtract intercept beta value from intercept + coefficient beta value to obtain the adjusted delta beta
nrow(male_res[abs(male_res$adjDB)>=0.03 & male_res$adj.P.Val<=0.05,]) #0

# Females
design_prs_female <- model.matrix(~ 1 + SCORE1_AVG + age + geno_V3 + geno_V4 + geno_V5 + CTP_PC1 + CTP_PC2 + CTP_PC3 + CTP_PC4 + CTP_PC5 + CTP_PC6 + plate + SentrixPosition_A + PD, data = meta_sub[meta_sub$sex=="F",])
prs_fit_female <- lmFit(mvals[,match(unlist(meta_sub[meta_sub$sex=="F","patient"]), colnames(mvals))], design_prs_female)
prs_fit_female <- eBayes(prs_fit_female)
# transform adjusted M-values to adjusted beta-values (code from Kruppa et al. 2021)
prs_fit_coef <- prs_fit_female$coefficients
m0_eff <- prs_fit_coef[, "(Intercept)"] #M-value model intercept
m1_eff <- m0_eff + prs_fit_coef[, "SCORE1_AVG"] #M-value model intercept plus GRS coefficient (delta M)

female_res <- limma::topTable(prs_fit_female, coef = 2, number = Inf, genelist=rownames(mvals))
female_res$adjDB <- m2beta(m1_eff) - m2beta(m0_eff) #transform each adjusted M-value to a beta-value, and subtract intercept beta value from intercept + coefficient beta value to obtain the adjusted delta beta
nrow(female_res[abs(female_res$adjDB)>=0.03 & female_res$adj.P.Val<=0.05,]) #0

# PD status EWAS check
design_pd <- model.matrix(~ 1 + PD + age + sex + geno_V3 + geno_V4 + geno_V5 + CTP_PC1 + CTP_PC2 + CTP_PC3 + CTP_PC4 + CTP_PC5 + CTP_PC6 + plate + SentrixPosition_A, data = meta_sub)
pd_fit <- lmFit(mvals, design_pd)
pd_fit <- eBayes(pd_fit)
# transform adjusted M-values to adjusted beta-values (code from Kruppa et al. 2021)
pd_fit_coef <- pd_fit$coefficients
m0_eff <- pd_fit_coef[, "(Intercept)"] #M-value model intercept
m1_eff <- m0_eff + pd_fit_coef[, "PD1"] #M-value model intercept plus GRS coefficient (delta M)

pd_res <- limma::topTable(pd_fit, coef = 2, number = Inf, genelist=rownames(mvals))
pd_res$adjDB <- m2beta(m1_eff) - m2beta(m0_eff) #transform each adjusted M-value to a beta-value, and subtract intercept beta value from intercept + coefficient beta value to obtain the adjusted delta beta
nrow(pd_res[abs(pd_res$adjDB)>=0.03 & pd_res$adj.P.Val<=0.05,]) #0

# save everything
# put limma results in separate csv's and together as one RData object for ease later
save(mvals, file="~/PD_GRS/TERRE_Mvals.RData")
write.csv(meta_sub, file="~/PD_GRS/meta_for_GRS_EWAS.csv", row.names=F)
save(pd_res, prs_res, control_res, case_res, male_res, female_res, file="~/PD_GRS/EWAS_TERRE/EWAS_limma_all.RData")
write.csv(pd_res, file="~/PD_GRS/EWAS_TERRE/EWAS_PD.csv", row.names=F)
write.csv(prs_res, file="~/PD_GRS/EWAS_TERRE/EWAS_GRS_cross_sex.csv", row.names=F)
write.csv(case_res, file="~/PD_GRS/EWAS_TERRE/EWAS_GRS_cases.csv", row.names=F)
write.csv(control_res, file="~/PD_GRS/EWAS_TERRE/EWAS_GRS_controls.csv", row.names=F)
write.csv(male_res, file="~/PD_GRS/EWAS_TERRE/EWAS_GRS_males.csv", row.names=F)
write.csv(female_res, file="~/PD_GRS/EWAS_TERRE/EWAS_GRS_females.csv", row.names=F)
```

### Plotting results

Thresholds: adjusted delta beta > 0.03 (exceeding RMSE of technical replicates), FDR < 0.05.

#### Cross-sex GRS EWAS
```{r volcano cross sex, eval=F}
meta_sub <- read.csv("~/PD_GRS/meta_for_GRS_EWAS.csv")
prs_res <- read.csv("~/PD_GRS/EWAS_TERRE/EWAS_GRS_cross_sex.csv")
load("~/PD_GRS/TERRE_Mvals.RData")

# annotate topTable
load("~/kobor_shared_coding_resource/EPIC_fdat.RData")
fdat <- fData_EPIC[match(rownames(mvals), fData_EPIC$TargetID),]
all(fdat$TargetID==rownames(mvals))
colnames(prs_res)[1] <- "TargetID"
prs_res <- left_join(prs_res, fdat, by="TargetID")

# logFC: delta M
# AveExpr: average M
colnames(prs_res)[2:3] <- c("delta_M", "avgM")

# plot with delta beta threshold
summary(prs_res$threshold_M <- (prs_res$adj.P.Val <= 0.05 & abs(prs_res$delta_M)>=1.5)) #24
summary(prs_res$threshold_B <- (prs_res$adj.P.Val <= 0.05 & abs(prs_res$adjDB)>=0.03)) #23
unique(prs_res[prs_res$threshold_B==TRUE,"threshold_M"]) #TRUE (all have delta M > 1.5)
prs_hits <- prs_res[prs_res$threshold_B==TRUE,]
write.csv(prs_hits, file="~/PD_GRS/EWAS_TERRE/EWAS_GRS_cross_sex_hits.csv", row.names=F)

max(prs_res[prs_res$threshold_B==TRUE,"P.Value"]) #1.432539e-06
prs_res$DNAm_change <- "NS"
prs_res[prs_res$threshold_B==TRUE & prs_res$adjDB>0,"DNAm_change"] <- "Increase"
prs_res[prs_res$threshold_B==TRUE & prs_res$adjDB<0,"DNAm_change"] <- "Decrease"
summary(as.factor(prs_res$DNAm_change))
#Decrease Increase       NS 
#      12       11   803754

# check gene names of hits
View(prs_hits[,c("TargetID","UCSC_REFGENE_NAME")])
prs_hits$UCSC_REFGENE_NAME
# replace repeated gene names
prs_hits$UCSC_REFGENE_NAME <- as.character(prs_hits$UCSC_REFGENE_NAME)
prs_hits$UCSC_REFGENE_NAME <- c("MAPT","KIAA1267","","ARHGAP27","KANSL1","","KANSL1","","MGC57346-CRHR1","MAPT",
                               "MAPT","","MAPT","","ARHGAP27","","KANSL1","KANSL1","","MAPT","ARHGAP27","MAPT","MGC57346-CRHR1")
prs_res_hits <- prs_res[match(prs_hits$TargetID, prs_res$TargetID),]
all(prs_res_hits$TargetID==prs_hits$TargetID)
prs_res_hits$UCSC_REFGENE_NAME <- prs_hits$UCSC_REFGENE_NAME
prs_res_other <- prs_res[-(which(prs_res$TargetID %in% prs_hits$TargetID)),]
prs_res_other$UCSC_REFGENE_NAME <- as.character(prs_res_other$UCSC_REFGENE_NAME)
prs_res <- rbind(prs_res_hits, prs_res_other)
write.csv(prs_res, file="~/PD_GRS/EWAS_TERRE/EWAS_GRS_cross_sex.csv", row.names=F)
write.csv(prs_res_hits, file="~/PD_GRS/EWAS_TERRE/EWAS_GRS_cross_sex_hits.csv", row.names=F)

max(abs(prs_res$delta_M)) #195.2651
max(-log10(prs_res$P.Value)) #11.95357

png("~/PD_GRS/volcano_GRS_cross_sex_Mvals.png", width=200, height=250)
ggplot(prs_res[,c("delta_M","P.Value","DNAm_change","UCSC_REFGENE_NAME")], aes(x=delta_M, y=-log10(P.Value), colour=DNAm_change)) +
  geom_point(size=1.75) +
  labs(legend.position = "none") +
  xlab("Adjusted Delta M") + ylab("-log10 P-Value") + theme_classic() + scale_color_manual(values=c("black","black","gray70")) + geom_hline(yintercept=-log10(1.670565e-06)) + geom_vline(xintercept=-1.5) + geom_vline(xintercept=1.5) +
    geom_label_repel(data = prs_res[prs_res$threshold==TRUE,c("delta_M","P.Value","DNAm_change","UCSC_REFGENE_NAME")],
            aes(label = UCSC_REFGENE_NAME), size = 4, force = 10, max.overlaps=8) + theme(legend.position="none")+ ylim(c(0,15)) + xlim(c(-200,200))
dev.off()

# Manhattan plot
prs_res$CHR <- as.factor(prs_res$CHR)
prs_res$CHR <- reorder.factor(prs_res$CHR, new.order=c(as.character(seq(1:22)), "X","Y"))
prs_res$col <- as.character(prs_res$CHR)
prs_res[prs_res$threshold_M==TRUE,"col"] <- "sig"
prs_res$col <- factor(prs_res$col, levels=c(as.character(seq(1:22)), "X","Y","sig"))
unique(prs_res$col)
prs_res <- prs_res %>% arrange(col)

png("~/PD_GRS/EWAS_TERRE/cross_sex_manhattan.png", width=613, height=218)
ggplot(prs_res, aes(x=CHR, y=-log10(P.Value), col=col)) + geom_jitter() + theme_classic() + scale_colour_manual(values=c(rep(c("black","grey50"),12),"red")) + theme(legend.position="none") + geom_hline(yintercept=-log10(1.670565e-06), linetype="dashed")+
    geom_label_repel(data = prs_res[prs_res$threshold_B==TRUE,],
            aes(label = UCSC_REFGENE_NAME), size = 3, force = 10, max.overlaps=15)
dev.off()

# chromosome track plot
# Chromosome Track
chr17_itrack <- IdeogramTrack(genome = "hg19", chromosome = "chr17")

# Basepair map track
chr17_gtrack <- GenomeAxisTrack(add53 = TRUE, add35 = TRUE, labelPos = "alternating")

#CpG track
#Each individual row is supposed to contain data (p-values) for a given sample (probe), where the coordinates for each single observation are constant across samples (probes)
#column: CpG
#row: sample
prs_sub <- prs_res[prs_res$CHR=="17" & prs_res$threshold_B==TRUE,]
prs_sub$log10P <- -log10(prs_sub$P.Value)
summary(prs_sub$MAPINFO)
max(prs_sub$MAPINFO) - min(prs_sub$MAPINFO) #840,684
prs_zoom <- t(as.matrix(prs_sub[,c("TargetID","MAPINFO","log10P")]))
colnames(prs_zoom) <- prs_zoom[2,]
prs_zoom <- prs_zoom[3,]
prs_zoom <- as.numeric(prs_zoom)
names(prs_zoom) <- prs_sub$MAPINFO
prs_zoom <- prs_zoom[sort(names(prs_zoom))]
chr17_dTrack_mC <- DataTrack(data = prs_zoom, name = "Chr17 CpGs", start = prs_sub$MAPINFO, width=1, chromosome = "chr17", genome = "hg19", type = "p", cex = 1, alpha =1, cex.legend = 1)

#intron/exon track
#UCSC genome browser tracks for chr17:43,502,999-44,343,683
#see https://genome.ucsc.edu/cgi-bin/hgTables?hgta_doSchemaDb=hg18&hgta_doSchemaTable=knownGene, https://genome.ucsc.edu/cgi-bin/hgTables?hgsid=1173061381_UepaHnvaOKFZKMOV4o7DtcNUHGVa&hgta_doSchemaDb=chlSab2&hgta_doSchemaTable=ncbiRefSeqCurated  for column name keys
knownGene_track <- fread("~/PD_GRS/chr17.hg19.knownGene.csv") #178 transcripts
knownGene_track$width <- knownGene_track$V3-knownGene_track$V2
knownGene_sub <- knownGene_track[grep("canonical",knownGene_track$V26),c("V1","V2","V3","V6","width","V18","V12","V11","V4","V20")]
colnames(knownGene_sub) <- c("chromosome","start","end","strand","width","gene","exonstarts","exonwidth","transcript","feature")
knownGene_sub <- knownGene_sub[-(grep("ENSG",knownGene_sub$gene)),]
knownGene_sub <- knownGene_sub[-(grep("pseudogene",knownGene_sub$feature)),]
knownGene_sub <- knownGene_sub[-(grep("misc_RNA",knownGene_sub$feature)),]
knownGene_sub <- knownGene_sub[-(grep("snRNA",knownGene_sub$feature)),]
knownGene_sub <- knownGene_sub[-(grep("lncRNA",knownGene_sub$feature)),]

grtrack <- GeneRegionTrack(knownGene_sub, genome = "hg19",
                           chromosome = "chr17", name = "Genes",
                           rstarts=knownGene_sub$exonstarts, rwidths=knownGene_sub$exonwidth,
                           transcript=knownGene_sub$transcript, feature=knownGene_sub$feature,
                           geneSymbol=TRUE, showId=TRUE, symbol=knownGene_sub$gene)

# plot tracks: with p-value summary
plotTracks(list(chr17_itrack, chr17_gtrack, grtrack, chr17_dTrack_mC), sizes = c(0.01,0.02,0.02,0.06), col.axis="black", background.title="white", from=min(prs_sub$MAPINFO), to=max(prs_sub$MAPINFO))

### DNAm tracks: beta values by GRS quartiles
betas <- getBeta(PD_RG_filtered)
betas_sub <- betas[rownames(betas) %in% prs_sub$TargetID,]
betas.mC <- as.data.frame(t(as.matrix(betas_sub)))

#categorize by GRS quartile
grs_quartiles <- summary(meta_sub$SCORE1_AVG)
meta_sub$grs_quartile <- "1st"
meta_sub[meta_sub$SCORE1_AVG>-0.004574 & meta_sub$SCORE1_AVG<=0.001083,"grs_quartile"] <- "2nd"
meta_sub[meta_sub$SCORE1_AVG>0.001083,"grs_quartile"] <- "3rd"
summary(as.factor(meta_sub$grs_quartile))
#1st 2nd 3rd 
# 55 108  55 

betas.mC <- betas.mC[match(meta_sub$patient, rownames(betas.mC)),]
all(rownames(betas.mC)==meta_sub$patient) #TRUE
betas.mC$Group <- meta_sub$grs_quartile

betas.mC.q1 <- betas.mC[betas.mC$Group == "1st",]
betas.mC.q1 <- betas.mC.q1[complete.cases(betas.mC.q1),]
betas.mC.q1$Group <- NULL
betas.mC.q1 <- as.data.frame(t(betas.mC.q1))
betas.mC.q1$symbol <- rownames(betas.mC.q1)

betas.mC.q2 <- betas.mC[betas.mC$Group == "2nd",]
betas.mC.q2 <- betas.mC.q2[complete.cases(betas.mC.q2),]
betas.mC.q2$Group <- NULL
betas.mC.q2 <- as.data.frame(t(betas.mC.q2))
betas.mC.q2$symbol <- rownames(betas.mC.q2)

betas.mC.q3 <- betas.mC[betas.mC$Group == "3rd",]
betas.mC.q3 <- betas.mC.q3[complete.cases(betas.mC.q3),]
betas.mC.q3$Group <- NULL
betas.mC.q3 <- as.data.frame(t(betas.mC.q3))
betas.mC.q3$symbol <- rownames(betas.mC.q3)

Group_betas_mC <- join(betas.mC.q1, betas.mC.q2)
Group_betas_mC <- join(Group_betas_mC, betas.mC.q3)
rownames(Group_betas_mC) <- Group_betas_mC$symbol

MAPT_fdat <- prs_res[prs_res$UCSC_REFGENE_NAME==
                       "MAPT",c("TargetID","CHR","MAPINFO")]
colnames(MAPT_fdat)[1] <- "symbol"
MAPT_fdat$genome <- "hg19"
Group_betas_mC <- join(Group_betas_mC, MAPT_fdat)
rownames(Group_betas_mC) <- Group_betas_mC$symbol
colnames(Group_betas_mC)
col_ord <- c("symbol","CHR","MAPINFO","genome",meta_sub[meta_sub$grs_quartile=="1st","patient"],
                                     meta_sub[meta_sub$grs_quartile=="2nd","patient"],
                                     meta_sub[meta_sub$grs_quartile=="3rd","patient"])
Group_betas_mC <- Group_betas_mC[,match(col_ord, colnames(Group_betas_mC))]
colnames(Group_betas_mC)

#mC track: boxplot
Group_betas_mC <- Group_betas_mC[complete.cases(Group_betas_mC$CHR),]
MAPT_dTrack_mC_box <- DataTrack(data = Group_betas_mC[,5:ncol(Group_betas_mC)], name = "Beta Value", start = Group_betas_mC$MAPINFO, width = 1, chromosome = "17", genome = unique(Group_betas_mC$genome), groups = c(rep("Q1", 55), rep("Q2",108), rep("Q3",55)), type = c("boxplot","a"), col=c("pink", "red","darkred"), col.boxplotFrame="transparent",box.width = 4, box.ratio=0.8, cex.legend = 1, ylim=c(0,1))

# plot MAPT region
plotTracks(list(chr17_itrack, chr17_gtrack, grtrack, MAPT_dTrack_mC_box), sizes = c(0.01,0.02,0.01,0.08), col.axis="black", background.title="white", from=min(MAPT_fdat$MAPINFO-6000), to=max(MAPT_fdat$MAPINFO-70000))

### KANSL1
Group_betas_mC <- join(betas.mC.q1, betas.mC.q2)
Group_betas_mC <- join(Group_betas_mC, betas.mC.q3)
rownames(Group_betas_mC) <- Group_betas_mC$symbol

KANSL1_fdat <- prs_res[prs_res$UCSC_REFGENE_NAME==
                       "KANSL1",c("TargetID","CHR","MAPINFO")]
colnames(KANSL1_fdat)[1] <- "symbol"
KANSL1_fdat$genome <- "hg19"
Group_betas_mC <- join(Group_betas_mC, KANSL1_fdat)
rownames(Group_betas_mC) <- Group_betas_mC$symbol
colnames(Group_betas_mC)
col_ord <- c("symbol","CHR","MAPINFO","genome",meta_sub[meta_sub$grs_quartile=="1st","patient"],
                                     meta_sub[meta_sub$grs_quartile=="2nd","patient"],
                                     meta_sub[meta_sub$grs_quartile=="3rd","patient"])
Group_betas_mC <- Group_betas_mC[,match(col_ord, colnames(Group_betas_mC))]
colnames(Group_betas_mC)

#mC track: boxplot
Group_betas_mC <- Group_betas_mC[complete.cases(Group_betas_mC$CHR),]
KANSL1_dTrack_mC_box <- DataTrack(data = Group_betas_mC[,5:ncol(Group_betas_mC)], name = "Beta Value", start = Group_betas_mC$MAPINFO, width = 1, chromosome = "17", genome = unique(Group_betas_mC$genome), groups = c(rep("Q1", 55), rep("Q2",108), rep("Q3",55)), type = c("boxplot","a"), col=c("pink", "red","darkred"), col.boxplotFrame="transparent",box.width = 4, box.ratio=0.8, cex.legend = 1, ylim=c(0,1))

# plot tracks
plotTracks(list(chr17_itrack, chr17_gtrack, grtrack, KANSL1_dTrack_mC_box), sizes = c(0.02,0.02,0.02,0.1), col.axis="black", background.title="white", from=min(KANSL1_fdat$MAPINFO)-35000, to=max(KANSL1_fdat$MAPINFO)-60000)

# Compare with previous result - how many overlap?
SI <- read_xlsx("~/PD_GRS/Table SI.xlsx", skip=2)
SI_cross_sex <- SI[SI$`TERRE Sample`=="Cross-sex",]
length(prs_res_hits$TargetID[prs_res_hits$TargetID %in% SI_cross_sex$Probe]) #22

# join data together to compare effects and p-values from each analysis
SI_sub <- SI_cross_sex[,c("Probe","logFC","BH Adjusted P Value")]
colnames(SI_sub) <- c("TargetID", "delta_M_orig", "adj.P.Val.orig")
combine <- left_join(prs_res_hits, SI_sub, by="TargetID")
combine <- combine[complete.cases(combine$delta_M_orig),]
max(c(combine$delta_M, combine$delta_M_orig)) #193.8127
min(c(combine$delta_M, combine$delta_M_orig)) #-176.0213

png("~/PD_GRS/EWAS_TERRE/EWAS_cross_sex_Mval_comparison.png", width=200, height=225)
ggplot(combine, aes(x=delta_M, y=delta_M_orig)) + geom_point() + geom_smooth(method="lm", col="grey") + theme_bw() + xlab("New adjusted delta M") + ylab("Original adjusted delta M") + xlim(c(-200,200)) + ylim(c(-200,200)) + ggtitle("Cross-sex EWAS M-value \ncomparison") + stat_cor(label.y=150)
dev.off()
```
![Updated EWAS model volcano](/home1/NEURO/schaffner/PD_GRS/EWAS_TERRE/volcano_GRS_cross_sex_Mvals.png)

![Updated EWAS model Manhattan plot](/home1/NEURO/schaffner/PD_GRS/EWAS_TERRE/cross_sex_manhattan.png)


![Chromosome 17 region vs -log10 P-values](/home1/NEURO/schaffner/PD_GRS/EWAS_TERRE/Gviz_chr17.png)


Most of the associations were also found previously, and many are on chromosome 17, mapping to the MAPT gene. This is consistent with Will's previous findings and demonstrative of the very strong association between the GRS and chr 17 DNA methylation.

![](/home1/NEURO/schaffner/PD_GRS/EWAS_TERRE/EWAS_cross_sex_Mval_comparison.png)

Adjusted delta M-values for the 22 overlapping probes are strongly correlated between the two methods.

#### Cross-sex PD status EWAS
```{r volcano pd, eval=F}
pd_res <- read.csv("~/PD_GRS/EWAS_TERRE/EWAS_PD.csv")

# annotate topTable
all(fdat$TargetID==rownames(mvals))
colnames(pd_res)[1] <- "TargetID"
pd_res <- left_join(pd_res, fdat, by="TargetID")

# logFC: delta M
# AveExpr: average M
colnames(pd_res)[2:3] <- c("delta_M", "avgM")

# plot with delta M threshold
summary(pd_res$threshold_M <- (pd_res$adj.P.Val <= 0.05 & abs(pd_res$delta_M)>=1.5)) #none pass threshold
min(pd_res$adj.P.Val) #0.9940453
pd_res$DNAm_change <- "NS"
write.csv(pd_res, file="~/PD_GRS/EWAS_PD.csv", row.names=F)

max(abs(pd_res$delta_M)) #1.526025
max(-log10(pd_res$P.Value)) #5.597638

# with the same plot parameters as above (for comparison)
png("~/PD_GRS/volcano_PD_Mvals.png", width=200, height=250)
ggplot(pd_res[,c("delta_M","P.Value","DNAm_change")], aes(x=delta_M, y=-log10(P.Value), colour=DNAm_change)) +
  geom_point(size=1.75) +
  labs(legend.position = "none") +
  xlab("Adjusted Delta M") + ylab("-log10 P-Value") + theme_classic() + scale_color_manual(values=c("gray70")) + geom_hline(yintercept=-log10(1.7e-06)) + geom_vline(xintercept=-1.5) + geom_vline(xintercept=1.5) + theme(legend.position="none")+ ylim(c(0,15)) + xlim(c(-200,200))
dev.off()

# zoomed in
png("~/PD_GRS/volcano_PD_Mvals_zoom.png", width=200, height=250)
ggplot(pd_res[,c("delta_M","P.Value","DNAm_change")], aes(x=delta_M, y=-log10(P.Value), colour=DNAm_change)) +
  geom_point(size=1.75) +
  labs(legend.position = "none") +
  xlab("Adjusted Delta M") + ylab("-log10 P-Value") + theme_classic() + scale_color_manual(values=c("gray70")) + geom_vline(xintercept=-1.5) + geom_vline(xintercept=1.5) + theme(legend.position="none")+ ylim(c(0,6)) + xlim(c(-2,2))
dev.off()
```
![Cross-sex EWAS for PD status](/home1/NEURO/schaffner/PD_GRS/EWAS_TERRE/volcano_PD_Mvals_zoom.png)

There are no DNAm changes related to PD status in the cross-sex sample (neither the effect sizes nor p-values come close to any thresholds). 


#### Cross-sex GRS EWAS in PD cases
```{r volcano cases, eval=F}
case_res <- read.csv("~/PD_GRS/EWAS_GRS_cases.csv")

# annotate topTable
all(fdat$TargetID==rownames(mvals))
colnames(case_res)[1] <- "TargetID"
case_res <- left_join(case_res, fdat, by="TargetID")

# logFC: delta M
# AveExpr: average M
colnames(case_res)[2:3] <- c("delta_M", "avgM")

# plot with delta M threshold
summary(case_res$threshold_M <- (case_res$adj.P.Val <= 0.05 & abs(case_res$delta_M)>=1.5)) #none
case_res$DNAm_change <- "NS"
write.csv(case_res, file="~/PD_GRS/EWAS_GRS_cases.csv", row.names=F)

max(abs(case_res$delta_M)) #318.9209
max(-log10(case_res$P.Value)) #5.01351

png("~/PD_GRS/volcano_GRS_cases_Mvals.png", width=200, height=250)
ggplot(case_res[,c("delta_M","P.Value","DNAm_change")], aes(x=delta_M, y=-log10(P.Value), colour=DNAm_change)) +
  geom_point(size=1.75) +
  labs(legend.position = "none") +
  xlab("Adjusted Delta M") + ylab("-log10 P-Value") + theme_classic() + scale_color_manual(values=c("gray70")) + geom_hline(yintercept=-log10(1.7e-06)) + geom_vline(xintercept=-1.5) + geom_vline(xintercept=1.5) + theme(legend.position="none")+ ylim(c(0,15)) + xlim(c(-350,350))
dev.off()
```
![EWAS in PD cases](/home1/NEURO/schaffner/PD_GRS/EWAS_TERRE/volcano_GRS_cases_Mvals.png)

There are no DNAm changes related to the GRS in PD cases (p-values are not significant, indicating high variability in effect sizes). 


#### Cross-sex GRS EWAS in controls
```{r volcano control, eval=F}
control_res <- read.csv("~/PD_GRS/EWAS_TERRE/EWAS_GRS_controls.csv")

# annotate topTable
all(fdat$TargetID==rownames(mvals))
colnames(control_res)[1] <- "TargetID"
control_res <- left_join(control_res, fdat, by="TargetID")

# logFC: delta M
# AveExpr: average M
colnames(control_res)[2:3] <- c("delta_M", "avgM")

# plot with delta M threshold
summary(control_res$threshold_M <- (control_res$adj.P.Val <= 0.05 & abs(control_res$delta_M)>=1.5)) #14
summary(control_res$threshold_B <- (control_res$adj.P.Val <= 0.05 & abs(control_res$adjDB)>=0.03)) #14
unique(control_res[control_res$threshold_B==TRUE,"threshold_M"]) #TRUE (all the same probes)
control_hits <- control_res[control_res$threshold_M==TRUE,]
write.csv(control_hits, file="~/PD_GRS/EWAS_TERRE/EWAS_GRS_controls_hits.csv", row.names=F)

max(control_res[control_res$threshold_M==TRUE,"P.Value"]) #6.823928e-07
control_res$DNAm_change <- "NS"
control_res[control_res$threshold_M==TRUE & control_res$delta_M>0,"DNAm_change"] <- "Increase"
control_res[control_res$threshold_M==TRUE & control_res$delta_M<0,"DNAm_change"] <- "Decrease"
summary(as.factor(control_res$DNAm_change))
#Decrease Increase       NS 
#       6        8   803763 

# check gene names of hits
View(control_hits[,c("TargetID","UCSC_REFGENE_NAME")])
control_hits$UCSC_REFGENE_NAME
# replace repeated gene names
control_hits$UCSC_REFGENE_NAME <- as.character(control_hits$UCSC_REFGENE_NAME)
control_hits$UCSC_REFGENE_NAME <- c("", "KANSL1", "ARGHAP27", "KANSL1", "MAPT", "KIAA1267", "", "", "", "MAPT", "MAPT", "SORCSC3", "MGC57346-CRHR1", "MAPT")
control_res_hits <- control_res[match(control_hits$TargetID, control_res$TargetID),]
all(control_res_hits$TargetID==control_hits$TargetID)
control_res_hits$UCSC_REFGENE_NAME <- control_hits$UCSC_REFGENE_NAME
control_res_other <- control_res[-(which(control_res$TargetID %in% control_hits$TargetID)),]
control_res_other$UCSC_REFGENE_NAME <- as.character(control_res_other$UCSC_REFGENE_NAME)
control_res <- rbind(control_res_hits, control_res_other)
write.csv(control_res, file="~/PD_GRS/EWAS_GRS_controls.csv", row.names=F)
write.csv(control_res_hits, file="~/PD_GRS/EWAS_GRS_controls_hits.csv", row.names=F)

max(abs(control_res$delta_M)) #218.4101
max(-log10(control_res$P.Value)) #9.334038

png("~/PD_GRS/volcano_GRS_controls_Mvals.png", width=200, height=250)
ggplot(control_res[,c("delta_M","P.Value","DNAm_change","UCSC_REFGENE_NAME")], aes(x=delta_M, y=-log10(P.Value), colour=DNAm_change)) +
  geom_point(size=1.75) +
  labs(legend.position = "none") +
  xlab("Adjusted Delta M") + ylab("-log10 P-Value") + theme_classic() + scale_color_manual(values=c("black","black","gray70")) + geom_hline(yintercept=-log10(6.823928e-07)) + geom_vline(xintercept=-1.5) + geom_vline(xintercept=1.5) +
    geom_label_repel(data = control_res[control_res$threshold==TRUE,c("delta_M","P.Value","DNAm_change","UCSC_REFGENE_NAME")],
            aes(label = UCSC_REFGENE_NAME), size = 4, force = 10, max.overlaps=6) + theme(legend.position="none")+ ylim(c(0,15)) + xlim(c(-250,250))
dev.off()

# How many overlap with combined case-control sample?
length(control_res_hits$TargetID[control_res_hits$TargetID %in% prs_res_hits$TargetID]) #13

# Look at the one that doesn't overlap
control_res_hits[-which(control_res_hits$TargetID %in% prs_res_hits$TargetID),c("TargetID","UCSC_REFGENE_NAME","UCSC_REFGENE_GROUP","delta_M","adj.P.Val")]
#     TargetID UCSC_REFGENE_NAME                              UCSC_REFGENE_GROUP   delta_M  adj.P.Val
#11 cg00112026              MAPT 5'UTR;5'UTR;5'UTR;5'UTR;5'UTR;5'UTR;5'UTR;5'UTR -40.04757 0.01990275

# Compare effect size with combined sample
# join data together to compare effects and p-values from each analysis
prs_sub <- prs_res_hits[,c("TargetID","delta_M","adj.P.Val")]
colnames(prs_sub) <- c("TargetID", "delta_M_combined", "adj.P.Val.combined")
combine <- left_join(control_res_hits, prs_sub, by="TargetID")
combine <- combine[complete.cases(combine$delta_M_combined),]
max(c(combine$delta_M, combine$delta_M_combined)) #218.4101
min(c(combine$delta_M, combine$delta_M_combined)) #-211.3666

png("~/PD_GRS/EWAS_control_vs_combined_Mvals.png", width=200, height=250)
ggplot(combine, aes(x=delta_M, y=delta_M_combined)) + geom_point() + geom_smooth(method="lm", col="grey") + theme_bw() + xlab("Controls adjusted delta M") + ylab("Combined case-control adjusted delta M") + xlim(c(-250,250)) + ylim(c(-250,250)) + ggtitle("Cross-sex EWAS M-value \ncomparison \n(controls vs. combined)") + stat_cor(label.y=150)
dev.off()

# Compare effect size with cases-only sample
# join data together to compare effects and p-values from each analysis
cases_sub <- case_res[,c("TargetID","delta_M","adj.P.Val")]
colnames(cases_sub) <- c("TargetID", "delta_M_cases", "adj.P.Val.cases")
combine <- left_join(control_res_hits, cases_sub, by="TargetID")
combine <- combine[complete.cases(combine$delta_M_cases),]
max(c(combine$delta_M, combine$delta_M_cases)) #218.4101
min(c(combine$delta_M, combine$delta_M_cases)) #-211.3666

png("~/PD_GRS/EWAS_control_vs_case_Mvals.png", width=200, height=250)
ggplot(combine, aes(x=delta_M, y=delta_M_cases)) + geom_point() + geom_smooth(method="lm", col="grey") + theme_bw() + xlab("Controls adjusted delta M") + ylab("Cases adjusted delta M") + xlim(c(-250,250)) + ylim(c(-250,250)) + ggtitle("Disease-stratified EWAS \nM-value comparison \n(controls vs. cases)") + stat_cor(label.y=150)
dev.off()
```
![EWAS in controls](/home1/NEURO/schaffner/PD_GRS/EWAS_TERRE/volcano_GRS_controls_Mvals.png)

14 probes are associated with the GRS in controls, 13 of which also appeared in the combined case-control sample.

![](/home1/NEURO/schaffner/PD_GRS/EWAS_TERRE/EWAS_control_vs_combined_Mvals.png)

The 14 probes have almost identical effect sizes in the control-only sample and the case-control combined sample adjusted for PD status.


![](/home1/NEURO/schaffner/PD_GRS/EWAS_TERRE/EWAS_control_vs_case_Mvals.png)

The 14 probes have a smaller effect size in PD cases than in controls.

#### Females GRS EWAS
```{r volcano female, eval=F}
female_res <- read.csv("~/PD_GRS/EWAS_GRS_females.csv")

# annotate topTable
colnames(female_res)[1] <- "TargetID"
female_res <- left_join(female_res, fdat, by="TargetID")

# logFC: delta M
# AveExpr: average M
colnames(female_res)[2:3] <- c("delta_M", "avgM")
write.csv(female_res, file="~/PD_GRS/EWAS_GRS_females.csv", row.names=F)

# plot with delta M threshold
summary(female_res$threshold_M <- (female_res$adj.P.Val <= 0.05 & abs(female_res$delta_M)>=1.5)) #no hits
min(female_res$adj.P.Val) #0.3603336
female_res$DNAm_change <- "NS"
write.csv(female_res, file="~/PD_GRS/EWAS_GRS_females.csv", row.names=F)

max(abs(female_res$delta_M)) #238.3878
max(-log10(female_res$P.Value)) #5.97757

png("~/PD_GRS/volcano_GRS_females_Mvals.png", width=200, height=250)
ggplot(female_res[,c("delta_M","P.Value","DNAm_change")], aes(x=delta_M, y=-log10(P.Value), colour=DNAm_change)) +
  geom_point(size=1.75) +
  labs(legend.position = "none") +
  xlab("Adjusted Delta M") + ylab("-log10 P-Value") + theme_classic() + scale_color_manual(values=c("gray70")) + geom_hline(yintercept=-log10(1.7e-06)) + geom_vline(xintercept=-1.5) + geom_vline(xintercept=1.5) + theme(legend.position="none")+ ylim(c(0,15)) + xlim(c(-250,250))
dev.off()

# Compare with previous result
# Check effect size of previously significant probes with effect size in the new analysis
SI_female <- SI[SI$`TERRE Sample`=="Female",]

# join data together to compare effects and p-values from each analysis
SI_sub <- SI_female[,c("Probe","logFC","BH Adjusted P Value")]
colnames(SI_sub) <- c("TargetID", "delta_M_orig", "adj.P.Val.orig")
female_res_sub <- female_res[female_res$TargetID %in% SI_sub$TargetID,] #41 probes
combine <- left_join(female_res_sub, SI_sub, by="TargetID")
combine <- combine[complete.cases(combine$delta_M_orig),]
max(c(combine$delta_M, combine$delta_M_orig)) #185.3679
min(c(combine$delta_M, combine$delta_M_orig)) #-186.6422

# M-values plot
png("~/PD_GRS/EWAS_female_Mval_comparison.png", width=200, height=225)
ggplot(combine, aes(x=delta_M, y=delta_M_orig)) + geom_point(col="lightpink4") + geom_smooth(method="lm", col="grey") + theme_bw() + xlab("New adjusted delta M") + ylab("Original adjusted delta M") + xlim(c(-200,200)) + ylim(c(-200,200)) + ggtitle("Female EWAS M-value \ncomparison") + stat_cor(label.y=150)
dev.off()

# P-values plot
png("~/PD_GRS/EWAS_female_pval_comparison.png", width=200, height=225)
ggplot(combine, aes(x=adj.P.Val, y=adj.P.Val.orig)) + geom_point(col="lightpink4") + geom_smooth(method="lm", col="grey") + theme_bw() + xlab("New adjusted p-value") + ylab("Original adjusted p-value") + xlim(c(0,1)) + ylim(c(0,1)) + ggtitle("Female EWAS p-value \ncomparison") + stat_cor(label.y=0.8)
dev.off()
```
![Updated female EWAS model](/home1/NEURO/schaffner/PD_GRS/EWAS_TERRE/volcano_GRS_females_Mvals.png)

Nothing reaches statistical significance for females when batch, cell type, and PD status are adjusted for.

![](/home1/NEURO/schaffner/PD_GRS/EWAS_TERRE/EWAS_female_Mval_comparison.png)

Adjusted delta M-values for the 41 probes originally discovered in the females EWAS are strongly correlated between the two methods.


![](/home1/NEURO/schaffner/PD_GRS/EWAS_TERRE/EWAS_female_pval_comparison.png)

Adjusted p-values for the 41 probes originally discovered in the females EWAS are all > 0.3 in the new EWAS.


#### Males GRS EWAS
```{r volcano male, eval=F}
male_res <- read.csv("~/PD_GRS/EWAS_GRS_males.csv")

# annotate topTable
colnames(male_res)[1] <- "TargetID"
male_res <- left_join(male_res, fdat, by="TargetID")

# logFC: delta M
# AveExpr: average M
colnames(male_res)[2:3] <- c("delta_M", "avgM")

# plot with delta M threshold
summary(male_res$threshold_M <- (male_res$adj.P.Val <= 0.05 & abs(male_res$delta_M)>=1.5)) #no hits
min(male_res$adj.P.Val) #0.1729346
min(male_res$P.Value) #2.151525e-07
male_res$DNAm_change <- "NS"
write.csv(male_res, file="~/PD_GRS/EWAS_GRS_males.csv", row.names=F)

max(abs(male_res$delta_M)) #187.229
max(-log10(male_res$P.Value)) #6.667254

png("~/PD_GRS/volcano_GRS_males_Mvals.png", width=200, height=250)
ggplot(male_res[,c("delta_M","P.Value","DNAm_change")], aes(x=delta_M, y=-log10(P.Value), colour=DNAm_change)) +
  geom_point(size=1.75) +
  labs(legend.position = "none") +
  xlab("Adjusted Delta M") + ylab("-log10 P-Value") + theme_classic() + scale_color_manual(values=c("gray70")) + geom_hline(yintercept=-log10(2.2e-07)) + geom_vline(xintercept=-1.5) + geom_vline(xintercept=1.5) + theme(legend.position="none")+ ylim(c(0,15)) + xlim(c(-250,250))
dev.off()

# Compare with previous result
# Check effect size of previously significant probes with effect size in the new analysis
SI_male <- SI[SI$`TERRE Sample`=="Male",]

# join data together to compare effects and p-values from each analysis
SI_sub <- SI_male[,c("Probe","logFC","BH Adjusted P Value")]
colnames(SI_sub) <- c("TargetID", "delta_M_orig", "adj.P.Val.orig")
male_res_sub <- male_res[male_res$TargetID %in% SI_sub$TargetID,] #59 probes
combine <- left_join(male_res_sub, SI_sub, by="TargetID")
combine <- combine[complete.cases(combine$delta_M_orig),]
max(c(combine$delta_M, combine$delta_M_orig)) #187.229
min(c(combine$delta_M, combine$delta_M_orig)) #-159.8757

# M-values plot
png("~/PD_GRS/EWAS_male_Mval_comparison.png", width=200, height=225)
ggplot(combine, aes(x=delta_M, y=delta_M_orig)) + geom_point(col="lightsteelblue4") + geom_smooth(method="lm", col="grey") + theme_bw() + xlab("New adjusted delta M") + ylab("Original adjusted delta M") + xlim(c(-200,200)) + ylim(c(-200,200)) + ggtitle("Male EWAS M-value \ncomparison") + stat_cor(label.y=150)
dev.off()

# P-values plot
png("~/PD_GRS/EWAS_male_pval_comparison.png", width=200, height=225)
ggplot(combine, aes(x=adj.P.Val, y=adj.P.Val.orig)) + geom_point(col="lightsteelblue4") + geom_smooth(method="lm", col="grey") + theme_bw() + xlab("New adjusted p-value") + ylab("Original adjusted p-value") + xlim(c(0,1)) + ylim(c(0,1)) + ggtitle("Male EWAS p-value \ncomparison") + stat_cor(label.y=0.8)
dev.off()
```
![Updated male EWAS model](/home1/NEURO/schaffner/PD_GRS/EWAS_TERRE/volcano_GRS_males_Mvals.png)

Nothing reaches statistical significance for males when batch, cell type, and PD status are adjusted for.

![](/home1/NEURO/schaffner/PD_GRS/EWAS_TERRE/EWAS_male_Mval_comparison.png)

Adjusted delta M-values for the 59 probes originally discovered in the males EWAS are strongly correlated between the two methods.


![](/home1/NEURO/schaffner/PD_GRS/EWAS_TERRE/EWAS_male_pval_comparison.png)

Adjusted p-values for the 59 probes originally discovered in the males EWAS are all > 0.2 in the new EWAS.

## Summary and next steps

In the EWAS models adjusted for batch, cell type, and disease status, some of the same hits show up in cross-sex analysis either with combined cases and controls (25 probes) or controls only (13 probes). These include the MAPT and KANSL1 loci which had very strong effect sizes. In sex-stratified analyses, nothing passes adjusted p-value cutoffs. The reduced number of hits likely means some probes initially discovered were associated with disease status, batch, or cell type.


Sample breakdown:

218 total

71 PD cases

147 controls

100 females

118 males


It is possible the sample size is too small in the sex-stratified and case-only analyses to detect effects. This can be checked with permutations and/or data reduction (e.g. subsetting to variable probes).

