---
title: "TERRE GRS: Epigenome-wide asssociation analysis"
output: html_notebook
---

The manuscript "A Parkinsonâ€™s disease genetic risk score associates with blood DNAm on chromosome 17" explores how a genetic risk score (GRS) for Parkinson's disease (PD) associates with blood DNA methylation in the TERRE study of French agricultural workers, and evaluates GRS associations with DNAm in each sex as well as their sensitivity to non-genetic factors. This was originally spearheaded by Will Casazza and was posted as a pre-print on medRxiv in 2023. The current revision (2024, led by Sam Schaffner) aims to address previous reviewer feedback and prepare the manuscript for re-submission.


In this script, EWAS is conducted to identify DNAm patterns associated with the GRS in the cross-sex and sex-stratified samples in TERRE. The models are adjusted for batch, cell type, and NOT disease status. This is to compare the effects of adjusting for batch and cell type alone with adjusting for disease status, particularly in sex-stratified samples.


Last updated June 21, 2024 by Samantha Schaffner

Based on code by Will Casazza

```{r setup, include=FALSE, eval=F}
library(lumi)
library(limma)
library(minfi)
library(rcompanion)
library(readxl)
library(tidyverse)
library(ggpubr)
library(glue)
library(data.table)
library(car)
library(bacon)
library(ggplot2)
library(ggrepel)
library(dplyr)
library(IlluminaHumanMethylationEPICanno.ilm10b4.hg19)
source("~/kobor_shared_coding_resource/qqplot_NG.R")
knitr::opts_chunk$set(echo = TRUE)
```


### Loading in and preparing data
```{r load dat, eval=FALSE}
meta <- read.csv("~/PD_GRS/meta_geno_DNAm_PCs.csv")

# DNAm data (2022, most recent QC'ed, same as npj PD 2024 article)
load("~/SHARE_DECIPHER/processed_DNAm_data/2022/TERRE_processed_2022/1-TERRE_RG_filtered.RData")
mvals <- getM(PD_RG_filtered)
```

# Association analysis

## EWAS model checks

Prior to looking at results, I'll assess whether the new EWAS model seems appropriate using three measures:

*1. Variance inflation factor*: The variance inflation factor (VIF) can be used to detect collinearity. Higher VIF values indicate a greater degree of collinearity. Typically VIF < 5 is considered acceptable.


*2. Q-Q plot of nominal p-values*: If the variable of interest does not affect global DNAm, p-values should distribute close to the center line on a normal Q-Q plot, with some deviation at the tail indicating associations detected. The shape of the Q-Q plot can visually indicate model inflation or deflation.


*3. Bias and inflation calculations*: The "bacon" package (van Iterson et al., 2017) was developed to calculate and optionally, correct for, bias and inflation observed in epigenome-wide association studies. It models test statistics using a Bayesian mixture model, and estimates the model parameters: bias (mean) and inflation (standard deviation). The Bayesian mixture model is proposed to more accurately model the quantitative nature and technical/biological influences on epigenomic and transcriptomic data, as opposed to the traditional genomic inflation factor (based on discrete genotypes).


```{r ewas models, eval=F}
## Preparing data
# match meta data and M-values
meta <- meta[complete.cases(meta$CTP_PC1),]
mvals <- mvals[,match(meta$patient, colnames(mvals))]
all(colnames(mvals)==meta$patient) #TRUE

# subset meta data to relevant columns and re-format
meta_sub <- meta[,c("patient","SCORE1_AVG","PD","age","sex","geno_V3","geno_V4","geno_V5","CTP_PC1","CTP_PC2","CTP_PC3","CTP_PC4","CTP_PC5","CTP_PC6","smoking","pesticides","head_trauma","alcohol1","levodopa","agonist","plate","SentrixPosition_A")]
str(meta_sub)
meta_sub$PD <- as.factor(meta_sub$PD)
meta_sub$sex <- as.factor(meta_sub$sex)
meta_sub$smoking <- as.factor(meta_sub$smoking)
meta_sub$pesticides <- as.factor(meta_sub$pesticides)
meta_sub$head_trauma <- as.factor(meta_sub$head_trauma)
meta_sub$alcohol1 <- as.factor(meta_sub$alcohol1)
meta_sub$levodopa <- as.factor(meta_sub$levodopa)
meta_sub$plate <- as.factor(meta_sub$plate)
meta_sub$SentrixPosition_A <- as.factor(meta_sub$SentrixPosition_A)

## Variance inflation factor
# check for one CpG (result is the same as when applied to all CpGs and averaged)
vif(lm(mvals[1,] ~ meta_sub$SCORE1_AVG + meta_sub$age + meta_sub$sex + meta_sub$geno_V3 + meta_sub$geno_V4 + meta_sub$geno_V5 + meta_sub$CTP_PC1 + meta_sub$CTP_PC2 + meta_sub$CTP_PC3 + meta_sub$CTP_PC4 + meta_sub$CTP_PC5 + meta_sub$CTP_PC6 + meta_sub$plate + meta_sub$SentrixPosition_A))[,3]

#       meta_sub$SCORE1_AVG               meta_sub$age               meta_sub$sex           meta_sub$geno_V3 
#                  1.026549                   1.104294                   1.537904                   1.076803 
#          meta_sub$geno_V4           meta_sub$geno_V5           meta_sub$CTP_PC1           meta_sub$CTP_PC2 
#                  1.098240                   1.065708                   1.092076                   1.114728 
#          meta_sub$CTP_PC3           meta_sub$CTP_PC4           meta_sub$CTP_PC5           meta_sub$CTP_PC6 
#                  1.307137                   1.139920                   1.243985                   1.156522 
#            meta_sub$plate meta_sub$SentrixPosition_A 
#                  1.134348                   1.049478 


## Q-Q plot
pvals <- sapply(1:nrow(mvals), function(x) {
  coef(summary(lm(mvals[x,] ~ meta_sub$SCORE1_AVG + meta_sub$age + meta_sub$sex + meta_sub$geno_V3 + meta_sub$geno_V4 + meta_sub$geno_V5 + meta_sub$CTP_PC1 + meta_sub$CTP_PC2 + meta_sub$CTP_PC3 + meta_sub$CTP_PC4 + meta_sub$CTP_PC5 + meta_sub$CTP_PC6 + meta_sub$plate + meta_sub$SentrixPosition_A)))[2,4]
   })
qqunif.plot(pvals, title="Model adjusted for batch and cell type")
hist(pvals)

## Bias and inflation (based on Z-values)
zvals <- qnorm(pvals)
bacon(zvals)
#...estimated bias: -0.12.
#...estimated inflation: 1.
```
![Q-Q plot for updated EWAS model](/home1/NEURO/schaffner/PD_GRS/qqplot_no_PD.png)

![](/home1/NEURO/schaffner/PD_GRS/pval_hist_no_PD.png)

Without PD status as a covariate, the Q-Q plot tail has a bit of an odd shape, and the p-value distribution is close to appropriate but less "null" than it should be. Bacon indicates a slight bias (-0.12) of test statistics, and no inflation.

Continue with differential methylation to examine the effect not adjusting for PD status has on results.

## Differential methylation analysis

Will used limma on all EPIC probes passing QC, modeling the data with M-values. I'll continue with this approach and transform results back to beta values later for plotting and interpretation purposes.

First, as Will did, I'll apply the updated models to (1) the cross-sex sample, (2) males, and (3) females. This permits us to detect whether GRS associations with DNAm are dependent on sex.

```{r prs ewas, eval=F}
# set up meta data
str(meta_sub[,c("SCORE1_AVG", "age", "sex", "geno_V3", "geno_V4", "geno_V5", "CTP_PC1", "CTP_PC2", "CTP_PC3", "CTP_PC4", "CTP_PC5", "CTP_PC6", "plate", "SentrixPosition_A")]) #checking format again - good

# limma design matrix and linear model fit
design_prs <- model.matrix(~ 1 + SCORE1_AVG + age + sex + geno_V3 + geno_V4 + geno_V5 + CTP_PC1 + CTP_PC2 + CTP_PC3 + CTP_PC4 + CTP_PC5 + CTP_PC6 + plate + SentrixPosition_A, data = meta_sub)
prs_fit <- lmFit(mvals, design_prs)
prs_fit <- eBayes(prs_fit)

# Males
design_prs_male <- model.matrix(~ 1 + SCORE1_AVG + age + geno_V3 + geno_V4 + geno_V5 + CTP_PC1 + CTP_PC2 + CTP_PC3 + CTP_PC4 + CTP_PC5 + CTP_PC6 + plate + SentrixPosition_A, data = meta_sub[meta_sub$sex=="M",])
prs_fit_male <- lmFit(mvals[,match(unlist(meta_sub[meta_sub$sex=="M","patient"]), colnames(mvals))], design_prs_male)
# Coefficients not estimable: plate6 plate9 
prs_fit_male <- eBayes(prs_fit_male)

# Females
design_prs_female <- model.matrix(~ 1 + SCORE1_AVG + age + geno_V3 + geno_V4 + geno_V5 + CTP_PC1 + CTP_PC2 + CTP_PC3 + CTP_PC4 + CTP_PC5 + CTP_PC6 + plate + SentrixPosition_A, data = meta_sub[meta_sub$sex=="F",])
prs_fit_female <- lmFit(mvals[,match(unlist(meta_sub[meta_sub$sex=="F","patient"]), colnames(mvals))], design_prs_female)
prs_fit_female <- eBayes(prs_fit_female)

#prs_res <- limma::topTable(prs_fit, coef = 2, number = Inf, genelist=rownames(mvals))
male_res <- limma::topTable(prs_fit_male, coef = 2, number = Inf, genelist=rownames(mvals))
female_res <- limma::topTable(prs_fit_female, coef = 2, number = Inf, genelist=rownames(mvals))

# save everything
# put limma results in separate csv's and together as one RData object for ease later
save(prs_res, male_res, female_res, file="~/PD_GRS/EWAS_limma_no_PD.RData")
#write.csv(prs_res, file="~/PD_GRS/EWAS_GRS_cross_sex_no_PD.csv", row.names=F)
write.csv(male_res, file="~/PD_GRS/EWAS_GRS_males_no_PD.csv", row.names=F)
write.csv(female_res, file="~/PD_GRS/EWAS_GRS_females_no_PD.csv", row.names=F)
```

### Plotting results

For now, I am plotting adjusted delta M-values (model coefficients) to compare with the previous result, and will use the same thresholds reported by Will (delta M > 1.5, padj < 0.05). After, I'll calculate the delta betas for the same model.

#### Cross-sex GRS EWAS
```{r volcano cross sex, eval=F}
# annotate topTable
load("~/kobor_shared_coding_resource/EPIC_fdat.RData")
fdat <- fData_EPIC[match(rownames(mvals), fData_EPIC$TargetID),]
all(fdat$TargetID==rownames(mvals))
colnames(prs_res)[1] <- "TargetID"
prs_res <- left_join(prs_res, fdat, by="TargetID")

# logFC: delta M
# AveExpr: average M
colnames(prs_res)[2:3] <- c("delta_M", "avgM")
write.csv(prs_res, file="~/PD_GRS/EWAS_GRS_cross_sex_no_PD.csv", row.names=F)

# plot with delta M threshold
summary(prs_res$threshold_M <- (prs_res$adj.P.Val <= 0.05 & abs(prs_res$delta_M)>=1.5)) #23
prs_hits <- prs_res[prs_res$threshold_M==TRUE,]
write.csv(prs_hits, file="~/PD_GRS/EWAS_GRS_cross_sex_no_PD_hits.csv", row.names=F)

max(prs_res[prs_res$threshold_M==TRUE,"P.Value"]) #6.375594e-07
prs_res$DNAm_change <- "NS"
prs_res[prs_res$threshold_M==TRUE & prs_res$delta_M>0,"DNAm_change"] <- "Increase"
prs_res[prs_res$threshold_M==TRUE & prs_res$delta_M<0,"DNAm_change"] <- "Decrease"
summary(as.factor(prs_res$DNAm_change))
#Decrease Increase       NS 
#       7       16   803754

# check gene names of hits
View(prs_hits[,c("TargetID","UCSC_REFGENE_NAME")])
prs_hits$UCSC_REFGENE_NAME
# replace repeated gene names
prs_hits$UCSC_REFGENE_NAME <- as.character(prs_hits$UCSC_REFGENE_NAME)
prs_hits$UCSC_REFGENE_NAME <- c("MAPT","KIAA1267","","ARHGAP27","","KANSL1","KANSL1","", "", "MGC57346-CRHR1", "MAPT", "MAPT", "", "MAPT", "KANSL1", "", "KANSL1", "ARHGAP27", "","MAPT","ARHGAP27", "", "MAPT")
prs_res_hits <- prs_res[match(prs_hits$TargetID, prs_res$TargetID),]
all(prs_res_hits$TargetID==prs_hits$TargetID)
prs_res_hits$UCSC_REFGENE_NAME <- prs_hits$UCSC_REFGENE_NAME
prs_res_other <- prs_res[-(which(prs_res$TargetID %in% prs_hits$TargetID)),]
prs_res_other$UCSC_REFGENE_NAME <- as.character(prs_res_other$UCSC_REFGENE_NAME)
prs_res <- rbind(prs_res_hits, prs_res_other)
write.csv(prs_res, file="~/PD_GRS/EWAS_GRS_cross_sex_no_PD.csv", row.names=F)
write.csv(prs_res_hits, file="~/PD_GRS/EWAS_GRS_cross_sex_no_PD_hits.csv", row.names=F)

max(abs(prs_res$delta_M)) #191.0527
max(-log10(prs_res$P.Value)) #11.85628

png("~/PD_GRS/volcano_GRS_cross_sex_no_PD_Mvals.png", width=200, height=250)
ggplot(prs_res[,c("delta_M","P.Value","DNAm_change","UCSC_REFGENE_NAME")], aes(x=delta_M, y=-log10(P.Value), colour=DNAm_change)) +
  geom_point(size=1.75) +
  labs(legend.position = "none") +
  xlab("Adjusted Delta M") + ylab("-log10 P-Value") + theme_classic() + scale_color_manual(values=c("black","black","gray70")) + geom_hline(yintercept=-log10(6.375594e-07)) + geom_vline(xintercept=-1.5) + geom_vline(xintercept=1.5) +
    geom_label_repel(data = prs_res[prs_res$threshold==TRUE,c("delta_M","P.Value","DNAm_change","UCSC_REFGENE_NAME")],
            aes(label = UCSC_REFGENE_NAME), size = 4, force = 10, max.overlaps=8) + theme(legend.position="none")+ ylim(c(0,15)) + xlim(c(-200,200))
dev.off()

# Compare with previous result - how many overlap?
## DNAm PCs model
SI <- read_xlsx("~/PD_GRS/Table SI.xlsx", skip=2)
SI_cross_sex <- SI[SI$`TERRE Sample`=="Cross-sex",]
length(prs_res_hits$TargetID[prs_res_hits$TargetID %in% SI_cross_sex$Probe]) #22
prs_res_hits[-which(prs_res_hits$TargetID %in% SI_cross_sex$Probe),c("TargetID","UCSC_REFGENE_NAME", "UCSC_REFGENE_GROUP", "delta_M","adj.P.Val")]
#     TargetID UCSC_REFGENE_NAME UCSC_REFGENE_GROUP  delta_M  adj.P.Val
#19 cg26278151                                      33.67215 0.01006582

## Model adjusted for PD, batch, and cell type
res_PD_adj <- read.csv("~/PD_GRS/EWAS_GRS_cross_sex_hits.csv")
length(prs_res_hits$TargetID[prs_res_hits$TargetID %in% res_PD_adj$TargetID]) #22
prs_res_hits[-which(prs_res_hits$TargetID %in% res_PD_adj$TargetID),c("TargetID","UCSC_REFGENE_NAME", "UCSC_REFGENE_GROUP", "delta_M","adj.P.Val")]
#     TargetID UCSC_REFGENE_NAME                  UCSC_REFGENE_GROUP  delta_M  adj.P.Val
#23 cg09764761              MAPT 3'UTR;3'UTR;3'UTR;3'UTR;3'UTR;3'UTR 34.64461 0.02228068
```
![EWAS model without adjustment for PD status](/home1/NEURO/schaffner/PD_GRS/volcano_GRS_cross_sex_no_PD_Mvals.png)

Most of the associations were also found previously, and many are on chromosome 17, mapping to the MAPT gene. This is consistent with Will's previous findings and demonstrative of the very strong association between the GRS and chr 17 DNA methylation.


#### Females GRS EWAS
```{r volcano female, eval=F}
# annotate topTable
colnames(female_res)[1] <- "TargetID"
female_res <- left_join(female_res, fdat, by="TargetID")

# logFC: delta M
# AveExpr: average M
colnames(female_res)[2:3] <- c("delta_M", "avgM")
write.csv(female_res, file="~/PD_GRS/EWAS_GRS_females_no_PD.csv", row.names=F)

# plot with delta M threshold
summary(female_res$threshold_M <- (female_res$adj.P.Val <= 0.05 & abs(female_res$delta_M)>=1.5)) #no hits
min(female_res$adj.P.Val) #0.4102435 - slightly higher than when adjusted for PD status
female_res$DNAm_change <- "NS"
write.csv(female_res, file="~/PD_GRS/EWAS_GRS_females_no_PD.csv", row.names=F)

max(abs(female_res$delta_M)) #205.3651
max(-log10(female_res$P.Value)) #6.061821

png("~/PD_GRS/volcano_GRS_females_no_PD_Mvals.png", width=200, height=250)
ggplot(female_res[,c("delta_M","P.Value","DNAm_change")], aes(x=delta_M, y=-log10(P.Value), colour=DNAm_change)) +
  geom_point(size=1.75) +
  labs(legend.position = "none") +
  xlab("Adjusted Delta M") + ylab("-log10 P-Value") + theme_classic() + scale_color_manual(values=c("gray70")) + geom_hline(yintercept=-log10(1.7e-06)) + geom_vline(xintercept=-1.5) + geom_vline(xintercept=1.5) + theme(legend.position="none")+ ylim(c(0,15)) + xlim(c(-250,250))
dev.off()
```
![Female EWAS model without adjustment for PD](/home1/NEURO/schaffner/PD_GRS/volcano_GRS_females_no_PD_Mvals.png)

Nothing reaches statistical significance for females when batch and cell type but not disease status are adjusted for.


#### Males GRS EWAS
```{r volcano male, eval=F}
# annotate topTable
colnames(male_res)[1] <- "TargetID"
male_res <- left_join(male_res, fdat, by="TargetID")

# logFC: delta M
# AveExpr: average M
colnames(male_res)[2:3] <- c("delta_M", "avgM")

# plot with delta M threshold
summary(male_res$threshold_M <- (male_res$adj.P.Val <= 0.05 & abs(male_res$delta_M)>=1.5)) #no hits
min(male_res$adj.P.Val) #0.2056077
min(male_res$P.Value) #2.846317e-07
male_res$DNAm_change <- "NS"
write.csv(male_res, file="~/PD_GRS/EWAS_GRS_males_no_PD.csv", row.names=F)

max(abs(male_res$delta_M)) #187.0451
max(-log10(male_res$P.Value)) #6.545717

png("~/PD_GRS/volcano_GRS_males_no_PD_Mvals.png", width=200, height=250)
ggplot(male_res[,c("delta_M","P.Value","DNAm_change")], aes(x=delta_M, y=-log10(P.Value), colour=DNAm_change)) +
  geom_point(size=1.75) +
  labs(legend.position = "none") +
  xlab("Adjusted Delta M") + ylab("-log10 P-Value") + theme_classic() + scale_color_manual(values=c("gray70")) + geom_hline(yintercept=-log10(2.2e-07)) + geom_vline(xintercept=-1.5) + geom_vline(xintercept=1.5) + theme(legend.position="none")+ ylim(c(0,15)) + xlim(c(-250,250))
dev.off()
```
![Male EWAS model without adjustment for PD status](/home1/NEURO/schaffner/PD_GRS/volcano_GRS_males_no_PD_Mvals.png)

Nothing reaches statistical significance for males when batch and cell type but not PD status are adjusted for.



## Summary

When PD status is not adjusted for, the cross-sex EWAS model is slightly biased. Fewer probes are statistically significant in the cross-sex EWAS than when PD status is adjusted for. Whether or not disease status is adjusted for, there are no hits in sex-stratified EWAS.

This indicates that adjusting for batch and cell type in models eliminated statistical significance in the sex-stratified samples, either due to sample size or previous confounding with batch/cell type (as opposed to confounding with disease status). Adjusting for batch and cell type also eliminates many hits in the cross-sex sample, while additionally adjusting for disease status improves the model further and reveals a few more hits in the cross-sex sample.





