---
title: "EWAS post-hoc SNP check"
output: html_notebook
---

The manuscript "A Parkinsonâ€™s disease genetic risk score associates with blood DNAm on chromosome 17" explores how a genetic risk score (GRS) for Parkinson's disease (PD) associates with blood DNA methylation in the TERRE study of French agricultural workers, and evaluates GRS associations with DNAm in each sex as well as their sensitivity to non-genetic factors. This was originally spearheaded by Will Casazza and was posted as a pre-print on medRxiv in 2023. The current revision aims to address previous reviewer feedback and prepare the manuscript for re-submission.

This script will test the associations of the GRS SNPs from Nalls et al. with the 11 individual CpGs discovered in TERRE EWAS and replicated in DIGPD EWAS.

Last updated Aug 24, 2025 by Samantha Schaffner: Double-checking whether any GRS-associated CpGs in the revised EWAS overlap with CpGs that colocalized with Nalls GWAS loci in Will's original analysis

### Libraries
```{r lib, eval=F}
library(data.table)
library(dplyr)
library(limma)
library(lumi)
library(readxl)
```

### Read in data
```{r dat, eval=F}
# GRS SNPs
risk_scores <- read.delim("~/SHARE_DECIPHER/pd_grs_ewas_integration/prs_analyses/pd_prs_file.txt", header=FALSE, sep=" ") 
# contains 107 SNPs from Nalls et al., prior to "excluding variants that 1) had a random-effects P value across all datasets > 4.67E-04 and 2) a conditional analysis P > 4.67E-04 using participant level 23andMe genotype data" (Nalls et al., 2019)
# 17 of these SNPs could be false positives or SNPs in LD with phenotype. Note to check this post-hoc.
# noting extra SNP list from Nalls et al. Table S2
excl_snps <- c("rs35643925","rs4954162","rs356228","rs356203","rs6875262","rs9267659","rs181609621","rs117073808","rs141128804",
"rs138017112","rs144755950","rs17686238","rs9912362","rs7221167","rs7225002","rs199453","rs2295545")
write.table(data.frame(SNP=excl_snps), file="~/PD_GRS/posthoc/excl_SNPs_Nalls.txt",sep="\t",row.names=F)

# need genotype data at each SNP here
genotype <- fread("~/SHARE_DECIPHER/pd_grs_ewas_integration/TERRE_QC/raw_data.imputed.r2_30.maf_mismatch.traw")
dim(genotype <- genotype[genotype$SNP %in% risk_scores$V1,]) #100 251
# 100 of the 107 SNPs are present in TERRE genotype data after imputation and QC

# replicated CpGs
repl_cpgs <- read.csv("~/PD_GRS/TERRE_cross_sex_in_DIGPD_B.csv")
repl_cpgs <- repl_cpgs[repl_cpgs$direction==TRUE,]
```

## EWAS with individual SNPs and CpGs
Testing each of the 107 SNPs for mQTL evidence with the GRS-associated CpGs, to see which SNPs are driving associations of the GRS with DNAm.

### In TERRE
```{r SNP EWAS, eval=F}
## Reading in meta data and beta values
meta <- read.csv("~/PD_GRS/meta_geno_DNAm_PCs.csv") # meta data frame used for EWAS
terre_meta <- fread("/home1/NEURO/SHARE_DECIPHER/terre_meta_master.csv") # meta data frame including IID
# subset meta data to the 218 individuals with matched DNAm and genotype data
dim(terre_meta_sub <- terre_meta[match(meta$patient, terre_meta$patient),]) # 218  69
meta <- left_join(meta, terre_meta_sub[,c("patient","IID")], by="patient")

# subset TERRE genotype data to those with matched DNAm data
all(meta$IID.y %in% colnames(genotype)[7:ncol(genotype)]) #TRUE
genotype <- as.data.frame(genotype) #so subsetting operation works
dim(genotype2 <- genotype[,c(colnames(genotype)[1:6], meta$IID.y)]) #100 225
all(colnames(genotype2)[7:ncol(genotype2)]==meta$IID.y) #TRUE
geno <- as.data.frame(t(as.matrix(genotype2[,7:ncol(genotype2)])))
colnames(geno) <- genotype2$SNP
meta_geno <- cbind(meta, geno)
write.csv(meta_geno, file="~/PD_GRS/meta_geno_GRS_SNPs.csv", row.names=F)

# DNAm data (2022, most recent QC'ed, same as npj PD 2024 article)
load("~/SHARE_DECIPHER/processed_DNAm_data/2022/TERRE_processed_2022/1-TERRE_RG_filtered.RData")
mvals <- getM(PD_RG_filtered)

## Preparing data

# subset M-values to replicated CpGs
dim(mvals <- mvals[rownames(mvals) %in% repl_cpgs$TargetID,]) #[1]  11 218

# subset meta data to relevant columns and re-format
meta_geno$PD <- as.factor(meta_geno$PD)
meta_geno$sex <- as.factor(meta_geno$sex)
meta_geno$smoking <- as.factor(meta_geno$smoking)
meta_geno$pesticides <- as.factor(meta_geno$pesticides)
meta_geno$head_trauma <- as.factor(meta_geno$head_trauma)
meta_geno$alcohol1 <- as.factor(meta_geno$alcohol1)
meta_geno$levodopa <- as.factor(meta_geno$levodopa)
meta_geno$plate <- as.factor(meta_geno$plate)
meta_geno$SentrixPosition_A <- as.factor(meta_geno$SentrixPosition_A)

# match meta data and M-values
meta_geno <- meta_geno[complete.cases(meta_geno$CTP_PC1),]
mvals <- mvals[,match(meta_geno$patient, colnames(mvals))]
all(colnames(mvals)==meta_geno$patient) #TRUE

####### Cross-sex PRS EWAS
# limma design matrix and linear model fit
mqtl_terre <- lapply(1:100, function(x){
  snp <- colnames(meta_geno)[167+x]
  print(paste(x, snp, sep=": "))
  design_prs <- model.matrix(~ 1 + meta_geno[,167+x] + age + sex + geno_V3 + geno_V4 + geno_V5 + CTP_PC1 + CTP_PC2 + CTP_PC3 + CTP_PC4 + CTP_PC5 + CTP_PC6 + plate + SentrixPosition_A + PD, data = meta_geno)
  prs_fit <- lmFit(mvals, design_prs)
  prs_fit <- eBayes(prs_fit)
  # transform adjusted M-values to adjusted beta-values (code from Kruppa et al. 2021)
  prs_fit_coef <- prs_fit$coefficients
  colnames(prs_fit_coef)[2] <- snp
  m0_eff <- prs_fit_coef[, "(Intercept)"] #M-value model intercept
  m1_eff <- m0_eff + prs_fit_coef[, snp] #M-value model intercept plus GRS coefficient (delta M)

  prs_res <- limma::topTable(prs_fit, coef = 2, number = Inf, genelist=rownames(mvals))
  prs_res$adjDB <- m2beta(m1_eff) - m2beta(m0_eff) #transform each adjusted M-value to a beta-value, and subtract intercept beta value from intercept + coefficient beta value to obtain the adjusted delta beta
  prs_res$snp <- snp
  return(prs_res)})
mqtl_terre <- do.call(rbind, mqtl_terre)
mqtl_terre$snp_excl <- (mqtl_terre$snp %in% excl_snps)
length(unique(mqtl_terre$snp)) #100
length(unique(mqtl_terre[mqtl_terre$snp_excl==TRUE,"snp"])) #14 of 17 excluded SNPs passed QC
length(unique(mqtl_terre[mqtl_terre$snp_excl==FALSE,"snp"])) #86 of 90 Nalls SNPs passed QC
write.csv(mqtl_terre, file="~/PD_GRS/posthoc/mqtl_terre_11x100.csv", row.names=FALSE)
  
nrow(mqtl_sig <- mqtl_terre[abs(mqtl_terre$adjDB)>=0.03 & mqtl_terre$adj.P.Val<=0.05,]) #63
length(unique(mqtl_sig$snp)) #8
terre_ind_snps <- unique(mqtl_sig$snp)
#"rs138017112" "rs17686238"  "rs62053943"  "rs9912362"   "rs7221167"   "rs7225002"   "rs199453"    "rs11658976" 
terre_ind_snps_sub <- unique(mqtl_sig[mqtl_sig$snp_excl==FALSE,"snp"]) #"rs62053943" "rs11658976"
```


#### Check TERRE mQTL status in other datasets

1. GoDMC database (Min et al., Nat Genet 2021): cis- and trans-mQTL detected in peripheral blood (450k array, n = ~30k, European-ancestry)

From GODMC README: "In our analysis we considered a cis pvalue smaller than 1e-8 and a trans pvalue smaller than 1e-14 as significant."

2. MuTHER (Hawe et al., Nat Genet 2022): cis- and trans-mQTL detected in peripheral blood (450k array, n = ~3k each European and South Asian ancestry)

```{r ext dat check terre, eval=F}
# Min et al. 2021 (GoDMC)
godmc <- fread("~/PD_GRS/coloc/godmc_assoc_meta_all_30Dec24.csv.gz")
#cistrans: cis=TRUE, Cis: Distance between SNP-CpG <1 MB
godmc <- godmc[(godmc$cistrans==TRUE & godmc$pval<1e-8)|(godmc$cistrans==FALSE & godmc$pval<1e-14),]

mqtl_sig$CpG_Min <- (mqtl_sig$ID %in% godmc$cpg)
summary(mqtl_sig$CpG_Min)
#   Mode   FALSE    TRUE 
#logical      56       7 

rm(godmc) #clearing large file to save space, load later as needed

hawe <- read.delim("~/PD_GRS/posthoc/hawe-cosmo-cpgs.txt", header=FALSE)
mqtl_sig$CpG_Hawe <- (mqtl_sig$ID %in% hawe$V1)
summary(mqtl_sig$CpG_Hawe)
#   Mode   FALSE    TRUE 
#logical      52      11 

table(mqtl_sig$CpG_Min, mqtl_sig$CpG_Hawe)
#        FALSE TRUE
#  FALSE    52    4
#  TRUE      0    7

write.csv(mqtl_sig, file="~/PD_GRS/posthoc/mqtl_terre_sig_anno.csv", row.names=FALSE)
```


#### Colocalization between TERRE mQTL CpGs and Nalls SNPs
See ~/PD_GRS/coloc/run_colocalization_SS.R
```{r coloc terre, eval=F}
coloc <- fread("~/PD_GRS/coloc/cross_mqtl_cross_sumstats_pd_snp_colocalization_ph4.txt.gz")
head(coloc)
#   nsnps    PP.H0.abf    PP.H1.abf    PP.H2.abf   PP.H3.abf    PP.H4.abf      probe locus locus_snp
#1:   228 4.718758e-29 9.904247e-01 2.415786e-31 0.005066008 4.509299e-03 cg00340179     1  rs356203
#2:   392 2.450009e-04 9.872223e-01 7.942090e-07 0.003190896 9.341020e-03 cg00379420     1  rs356203
#3:   196 1.879186e-43 5.492019e-35 3.421668e-09 0.999999989 7.337095e-09 cg00538060     1  rs356203
#4:   136 1.555653e-02 9.699517e-01 3.917218e-05 0.002430368 1.202221e-02 cg00907891     1  rs356203
#5:   527 9.896689e-46 1.469541e-22 6.734545e-24 1.000000000 8.221536e-13 cg01966878     1  rs356203
#6:   198 5.955894e-09 5.449637e-07 1.053193e-02 0.963644303 2.582322e-02 cg02262221     1  rs356203

#PP.H4.abf is the posterior probability of H4 being true (one common causal variant)
#WC considered pp > 90% to be strong evidence
nrow(coloc_sub <- coloc[coloc$PP.H4.abf>0.9 & coloc$locus_snp %in% mqtl_sig$snp,])  #14
length(unique(coloc_sub$locus)) #1 locus has evidence for a SNP being a shared causal variant for PD risk (in Nalls dataset) and TERRE mQTL
length(unique(coloc_sub$probe)) #this SNP is an mQTL for 14 CpGs

length(unique(mqtl_sig$ID)) #11
nrow(coloc_sub <- coloc_sub[coloc_sub$probe %in% mqtl_sig$ID,]) #the GRS SNP is an mQTL for 3 of the CpGs that came up in the cross-sex GRS EWAS

write.csv(coloc_sub, file="~/PD_GRS/coloc/terre_posthoc_coloc.csv", row.names=FALSE)
```

rs7225002 has evidence for being a shared causal variant for affecting PD risk in the Nalls dataset and affecting DNAm in TERRE at 3 CpGs: cg01341218, cg12609785, and cg15633388.

### In DIGPD
```{r DIGPD SNP EWAS, eval=F}
## Reading in meta data and beta values
meta <- read.csv("~/PD_GRS/digpd_meta_geno_DNAm_PCs.csv") # meta data frame used for EWAS

# need genotype data at each SNP here
genotype <- fread("~/SHARE_DECIPHER/Genetic_data/DIGPD_Processed/raw_data.imputed.r2_90.maf_mismatch.traw")
dim(genotype <- genotype[genotype$SNP %in% risk_scores$V1,]) #94 261
# 94 of the 107 SNPs are present in DIGPD genotype data after imputation and QC

# subset DIGPD genotype data to those with matched DNAm data
colnames(genotype) <-  gsub("_AOR_.*", "", colnames(genotype))
length(meta$IID[meta$IID %in% colnames(genotype)]) #222
genotype <- as.data.frame(genotype)
dim(genotype2 <- genotype[,c(colnames(genotype)[1:6], meta$IID)]) # 94 228
all(colnames(genotype2)[7:ncol(genotype2)]==meta$IID) #TRUE
geno <- as.data.frame(t(as.matrix(genotype2[,7:ncol(genotype2)])))
colnames(geno) <- genotype2$SNP
meta_geno <- cbind(meta, geno)
write.csv(meta_geno, file="~/PD_GRS/meta_geno_GRS_SNPs_digpd.csv", row.names=F)

summary(geno) #as cases and controls were genotyped on diff platforms, some SNPs have no genotype for cases or for controls...
#subset to complete genotypes for all subjects
ncol(geno2 <- geno[,apply(geno,2, function(col) all(complete.cases(col)))]) #68 SNPs complete for all subjects
summary(geno2)
meta_geno2 <- cbind(meta, geno2)
write.csv(meta_geno2, file="~/PD_GRS/meta_geno_GRS_SNPs_digpd_complete.csv", row.names=F)

# DNAm data (2022, most recent QC'ed, same as npj PD 2024 article)
rm(PD_RG_filtered) #removing TERRE object
load("~/SHARE_DECIPHER/processed_DNAm_data/2022/DIGPD_processed_2022/1-DIGPD_RG_filtered.RData")
mvals <- getM(DIGPD_RG_filtered)

## Preparing data

# subset M-values to replicated CpGs and 222 individuals with visit 1 DNAm data
dim(mvals <- mvals[rownames(mvals) %in% repl_cpgs$TargetID,]) #[1]  11 425
dim(mvals <- mvals[,match(meta_geno2$Sample_Name, colnames(mvals))]) #[1]  11 222

# subset meta data to relevant columns and re-format
meta_geno2$PD <- as.factor(meta_geno2$PD)
meta_geno2$reportedSex <- as.factor(meta_geno2$reportedSex)
meta_geno2$smoking <- as.factor(meta_geno2$smoking)
meta_geno2$pesticides <- as.factor(meta_geno2$pesticides)
meta_geno2$head_trauma <- as.factor(meta_geno2$head_trauma)
meta_geno2$alcohol <- as.factor(meta_geno2$alcohol)
meta_geno2$levodopa <- as.factor(meta_geno2$levodopa)
meta_geno2$plate <- as.factor(meta_geno2$plate)
meta_geno2$SentrixPosition_A <- as.factor(meta_geno2$SentrixPosition_A)

# match meta_geno2 data and M-values
meta_geno2 <- meta_geno2[complete.cases(meta_geno2$CTP_PC1),]
all(colnames(mvals)==meta_geno2$Sample_Name) #TRUE

####### Cross-sex PRS EWAS
# limma design matrix and linear model fit
mqtl_digpd <- lapply(1:68, function(x){
  snp <- colnames(meta_geno2)[71+x]
  print(paste(x, snp, sep=": "))
  design_prs <- model.matrix(~ 1 + meta_geno2[,71+x] + Age + reportedSex + GPC1 + GPC2 + GPC3 + CTP_PC1 + CTP_PC2 + CTP_PC3 + CTP_PC4 + plate + SentrixPosition_A + PD, data = meta_geno2)
  prs_fit <- lmFit(mvals, design_prs)
  prs_fit <- eBayes(prs_fit)
  # transform adjusted M-values to adjusted beta-values (code from Kruppa et al. 2021)
  prs_fit_coef <- prs_fit$coefficients
  colnames(prs_fit_coef)[2] <- snp
  m0_eff <- prs_fit_coef[, "(Intercept)"] #M-value model intercept
  m1_eff <- m0_eff + prs_fit_coef[, snp] #M-value model intercept plus GRS coefficient (delta M)

  prs_res <- limma::topTable(prs_fit, coef = 2, number = Inf, genelist=rownames(mvals))
  prs_res$adjDB <- m2beta(m1_eff) - m2beta(m0_eff) #transform each adjusted M-value to a beta-value, and subtract intercept beta value from intercept + coefficient beta value to obtain the adjusted delta beta
  prs_res$snp <- snp
  return(prs_res)})
mqtl_digpd <- do.call(rbind, mqtl_digpd)
mqtl_digpd$snp_excl <- (mqtl_digpd$snp %in% excl_snps)
length(unique(mqtl_digpd$snp)) #68
length(unique(mqtl_digpd[mqtl_digpd$snp_excl==TRUE,"snp"])) #7 of 17 excluded SNPs passed QC
length(unique(mqtl_digpd[mqtl_digpd$snp_excl==FALSE,"snp"])) #61 of 90 Nalls SNPs passed QC
write.csv(mqtl_digpd, file="~/PD_GRS/posthoc/mqtl_digpd_11x68.csv", row.names=FALSE)
  
nrow(mqtl_sig <- mqtl_digpd[abs(mqtl_digpd$adjDB)>=0.03 & mqtl_digpd$adj.P.Val<=0.05,]) #16
length(unique(mqtl_sig$snp)) #3
digpd_ind_snps <- unique(mqtl_sig$snp) #"rs9912362" "rs7225002" "rs199453" 
digpd_ind_snps_sub <- unique(mqtl_sig[mqtl_sig$snp_excl==FALSE,"snp"]) #none
```


#### Check DIGPD mQTL status in other datasets

1. GoDMC database (Min et al., Nat Genet 2021): cis- and trans-mQTL detected in peripheral blood (450k array, n = ~30k, European-ancestry)

From GODMC README: "In our analysis we considered a cis pvalue smaller than 1e-8 and a trans pvalue smaller than 1e-14 as significant."

2. MuTHER (Hawe et al., Nat Genet 2022): cis- and trans-mQTL detected in peripheral blood (450k array, n = ~3k each European and South Asian ancestry)

```{r ext dat check digpd, eval=F}
# Min et al. 2021 (GoDMC)
godmc <- fread("~/PD_GRS/coloc/godmc_assoc_meta_all_30Dec24.csv.gz")
#cistrans: cis=TRUE, Cis: Distance between SNP-CpG <1 MB
godmc <- godmc[(godmc$cistrans==TRUE & godmc$pval<1e-8)|(godmc$cistrans==FALSE & godmc$pval<1e-14),]

mqtl_sig$CpG_Min <- (mqtl_sig$ID %in% godmc$cpg)
summary(mqtl_sig$CpG_Min)
#   Mode   FALSE    TRUE 
#logical      14       2 

rm(godmc) #clearing large file to save space, load later as needed

mqtl_sig$CpG_Hawe <- (mqtl_sig$ID %in% hawe$V1)
summary(mqtl_sig$CpG_Hawe)
#   Mode   FALSE    TRUE 
#logical      13       3 

table(mqtl_sig$CpG_Min, mqtl_sig$CpG_Hawe)
#        FALSE TRUE
#  FALSE    13    1
#  TRUE      0    2

write.csv(mqtl_sig, file="~/PD_GRS/posthoc/mqtl_digpd_sig_anno.csv", row.names=FALSE)
```


#### Colocalization between DIG-PD mQTL CpGs and Nalls SNPs
See ~/PD_GRS/coloc/run_colocalization_DIGPD_SS_9Feb2025.R
```{r coloc digpd, eval=F}
coloc <- fread("~/PD_GRS/coloc/cross_mqtl_cross_sumstats_digpd_digpd_sig_pd_snp_colocalization_ph4.txt.gz")
head(coloc)
#   nsnps    PP.H0.abf    PP.H1.abf    PP.H2.abf    PP.H3.abf PP.H4.abf      probe locus  locus_snp
#1:     1 1.364976e-19 1.750143e-17 7.799166e-06 0.000000e+00 0.9999922 cg01341218     7 rs58879558
#2:     1 1.238733e-09 3.880626e-05 3.191972e-08 0.000000e+00 0.9999612 cg02478991     7 rs58879558
#3:     2 1.806078e-23 1.747428e-17 1.033791e-09 2.198163e-07 0.9999998 cg04226788     7 rs58879558
#4:     1 5.710106e-11 3.880626e-05 1.471382e-09 0.000000e+00 0.9999612 cg05314706     7 rs58879558
#5:     2 8.715584e-19 1.752031e-17 4.988760e-05 2.907241e-06 0.9999472 cg09793084     7 rs58879558
#6:     2 2.278026e-28 2.194478e-17 1.038072e-14 5.684342e-17 1.0000000 cg09860564     7 rs58879558

#PP.H4.abf is the posterior probability of H4 being true (one common causal variant)
#WC considered pp > 90% to be strong evidence
nrow(coloc_sub <- coloc[coloc$PP.H4.abf>0.9 & coloc$locus_snp %in% mqtl_sig$snp,])  #11
length(unique(coloc_sub$locus)) #1 locus has evidence for a SNP being a shared causal variant for PD risk (in Nalls dataset) and TERRE mQTL
length(unique(coloc_sub$probe)) #this SNP is an mQTL for 11 CpGs

length(unique(mqtl_sig$ID)) #11
nrow(coloc_sub <- coloc_sub[coloc_sub$probe %in% mqtl_sig$ID,]) #the GRS SNP is an mQTL for 11 of the CpGs that came up in the cross-sex GRS EWAS

write.csv(coloc_sub, file="~/PD_GRS/coloc/digpd_posthoc_coloc.csv", row.names=FALSE)
```

rs7225002 has evidence for being a shared causal variant for affecting PD risk in the Nalls dataset and affecting DNAm in DIG-PD at 11 CpGs, including cg01341218, cg12609785, and cg15633388 (the same 3 from TERRE), plus 8 more.

### Checking GRS-associated CpGs against colocalized CpGs from Nalls/GODMC analysis
Added Aug 24, 2025. Check for the 23 from initial TERRE EWAS as well as the 11 replicated in DIG-PD.

Also checking EWAS stats for 12 CpGs from GODMC colocalization analysis that were on chromosome 17.
```{r coloc godmc, eval=F}
terre_ewas <- read.csv("~/PD_GRS/EWAS_TERRE/EWAS_GRS_cross_sex_hits.csv")
coloc_godmc <- read_xlsx("~/PD_GRS/coloc_analysis_WC.xlsx", sheet=1, skip=1)
coloc_godmc$terre_grs_ewas <- (coloc_godmc$Probe %in% terre_ewas$TargetID)
coloc_godmc$digpd_repl <- (coloc_godmc$Probe %in% repl_cpgs$TargetID)
summary(coloc_godmc[,c("terre_grs_ewas","digpd_repl")])
# terre_grs_ewas  digpd_repl     
# Mode :logical   Mode :logical  
# FALSE:79        FALSE:79  

#check the EWAS stats for the 12 CpGs on chr17 that had mQTL colocalized with Nalls loci
coloc_probes <- coloc_godmc[coloc_godmc$Chromosome=="chr17","Probe"]
coloc_probes <- coloc_probes$Probe
terre_ewas <- read.csv("~/PD_GRS/EWAS_TERRE/EWAS_GRS_cross_sex.csv")
terre_ewas_sub <- terre_ewas[terre_ewas$TargetID %in% coloc_probes,c("TargetID","delta_M","P.Value","adj.P.Val","adjDB","CHR","MAPINFO","UCSC_REFGENE_NAME","UCSC_REFGENE_GROUP","RELATION_TO_UCSC_CPG_ISLAND","METHYL450_LOCI","threshold_B")]
terre_ewas_sub[terre_ewas_sub$P.Value<=0.05,c("TargetID","UCSC_REFGENE_NAME","P.Value","adj.P.Val","adjDB")]
#        TargetID UCSC_REFGENE_NAME    P.Value adj.P.Val      adjDB
#23603 cg15679139 LOC100128977;IMP5 0.02375372 0.8068494 0.05474954

write.csv(terre_ewas_sub, file="~/PD_GRS/posthoc/coloc_GODMC_in_TERRE.csv")
```

