---
title: "PRSice2 Development of risk score"
output: github_document
---

```{r setup, include=FALSE}
library(tidyverse)
library(ggrepel)
library(data.table)
library(knitr)
library(limma)
library(foreach)
library(doParallel)
knitr::opts_chunk$set(echo = TRUE)
```
# Step 0: Prepare covariates and input files
```{r}
IDs <- fread("~/genotype_qc/TERRE_QC/all_imputed_r2_30_rsid_hard_call.fam")[, .(FID = V1, IID = V2)]
covariate <- fread("../cis_mQTL_analyses/terre_data/covariates_10_methy_PC.txt")[id != "head_trauma_loc"]
covariate <- cbind(IDs, covariate %>% transpose(make.names = "id"))

PD <- fread("../cis_mQTL_analyses/terre_data/covariates_CTP_PD.txt")[id == "PD"]
PD <- cbind(IDs, PD %>% transpose(make.names = "id"))

head(covariate)
head(PD)
fwrite(PD, "TERRE.pheno", sep = "\t")
fwrite(covariate, "TERRE.covariate", sep = "\t")
```
```{r}
covariate
```


# Step 1: Run PRSice-2 on Nalls et al 2019 Sumstats
```{bash,eval=FALSE}
Rscript /home1/NEURO/casazza/PRSice.R \
    --prsice /home1/NEURO/casazza/PRSice_linux\
    --base /home1/NEURO/casazza/nalls_PD.QC.gz\
    --base-info INFO:0.8 \
    --base-maf MAF:0.01 \
    --cov TERRE.covariate \
    --beta  \
    --out TERRE_PRSice \
    -q 5\
    --all-score\
    --pheno TERRE.pheno \
    --snp SNP \
    --stat b \
    --pvalue p\
    --target /home1/NEURO/casazza/genotype_qc/TERRE_QC/all_imputed_r2_30_rsid_hard_call \
    --thread 32

```
# Step 2: Evaluate output
```{r, out.width="400px"}
include_graphics("TERRE_PRSice_BARPLOT_2021-11-02.png")
include_graphics("TERRE_PRSice_HIGH-RES_PLOT_2021-11-02.png")
include_graphics("TERRE_PRSice_QUANTILES_PLOT_2021-11-02.png")
```
## Plotting PRSice Data on my own
```{r}
prsice_meta <- fread("prsice_data/TERRE_PRSice.prsice")
ggplot(prsice_meta, aes(Threshold, R2, color = -log10(P))) +
  geom_point(size = 1) +
  labs(y = bquote("R"^2), x = "GWAS P-Value Threshold", color = bquote("-log"["10"] ~ "(P)")) +
  theme_minimal()
ggplot(prsice_meta[Threshold <= 0.5], aes(Threshold, Num_SNP, color = -log10(P))) +
  geom_point() +
  scale_y_continuous(breaks = c(seq(0, 1e5, 2.5e4), seq(2e5, 6e5, 1e5))) +
  theme_minimal()
```

# Step 3 run linear model at different thresholds for SNP inclusion
```{r}
load("/home1/NEURO/SHARE_DECIPHER/processed_DNAm_data/TERRE_processed_2021/betas_combat.RData") # betas_sub
# Assign genotyping ID to data
original_covars <- fread("/home1/NEURO/SHARE_DECIPHER/terre_meta_master.csv")[, .(patient, IID = gsub("_PAE.*", "", IID))]
colnames(betas_combat) <- original_covars$IID[match(colnames(betas_combat), original_covars$patient)]
betas_combat <- betas_combat[, colnames(betas_combat) %in% covariate$IID]
```
Let's check how the data looks for the first 5 subjects:
```{r}
ggplot(betas_combat[, 1:5] %>% as.data.table(keep.rownames = T) %>% melt(id.vars = "rn", value.name = "betas", variable.name = "subject"), aes(betas, color = subject)) +
  geom_density()
```

### Match DNA, PRS, and metadata
```{r}
prsice_all <- fread("prsice_data/TERRE_PRSice.all_score")[match(colnames(betas_combat), IID), .(FID, IID, `Pt_0.0219001`, `Pt_5e-08`, `Pt_5.005e-05`, `Pt_0.00010005`, `Pt_0.00100005`, `Pt_0.0101501`, `Pt_0.1`, `Pt_0.2`, `Pt_0.3`, `Pt_0.4`, `Pt_0.5`, `Pt_1`)]
covariate <- covariate[match(colnames(betas_combat), IID)]
all(covariate$IID == colnames(betas_combat))
all(covariate$IID == prsice_all$IID)
```
### Run limma

```{r}
mvalues <- lumi::beta2m(betas_combat)
prs_mat <- prsice_all[, -c(1, 2)]
cov_mat <- covariate[, -c(1, 2)]
```

```{r}
registerDoParallel(ncol(prs_mat) / 2)
hits <- foreach(prs_thresh = colnames(prs_mat)) %dopar% {
  design_prs <- model.matrix(~., data = cbind(prs_mat[, ..prs_thresh], cov_mat))
  prs_fit <- lmFit(mvalues, design_prs)
  prs_fit <- eBayes(prs_fit)
  topTable(prs_fit, coef = 2, adjust.method = "bonf", p.value = 0.05, number = Inf, genelist = rownames(mvalues))
}
names(hits) <- colnames(prs_mat)
hits_by_thresh_bonf <- rbindlist(hits, idcol = "threshold", fill = TRUE)
```
```{r}
ggplot(hits_by_thresh_bonf[, .(hits = .N), by = threshold] %>% mutate(threshold = recode_factor(threshold, `Pt_0.0219001` = "0.0219", `Pt_5e-08` = "5e-8", `Pt_5.005e-05` = "5e-5", `Pt_0.00010005` = "1e-4", `Pt_0.00100005` = "1e-3", `Pt_0.0101501` = "1e-2", `Pt_0.1` = "0.1", `Pt_0.2` = "0.2", `Pt_0.3` = "0.3", `Pt_0.4` = "0.4", `Pt_0.5` = "0.5", `Pt_1` = "1.0")), aes(threshold, hits, label = hits)) +
  geom_text(vjust = -0.25) +
  geom_col() +
  labs(x = "GWAS P Value Threshold", y = "EWAS Hits") +
  theme_minimal()
```

```{r}
library(missMethyl)
gometh(hits_by_thresh_bonf[threshold == "Pt_5e-08"]$ID, array.type = "EPIC", all.cpg = rownames(mvalues))
```

```{r}
top_design_prs <- model.matrix(~., data = cbind(prs_mat[, `Pt_5e-08`], cov_mat))
top_prs_fit <- lmFit(mvalues, top_design_prs)
top_prs_fit <- eBayes(top_prs_fit)
top_prs_hits <- topTable(top_prs_fit, coef = 2, adjust.method = "bonf", number = Inf, genelist = rownames(mvalues))

top_design_prs_gxe <- model.matrix(~., data = cbind(prs_mat[, `Pt_5e-08`], cov_mat))
top_prs_fit_gxe <- lmFit(mvalues, top_design_prs_gxe)
top_prs_fit_gxe <- eBayes(top_prs_fit_gxe)
top_prs_hits_gxe <- topTable(top_prs_fit_gxe, coef = 2, adjust.method = "bonf", number = Inf, genelist = rownames(mvalues))
```
```{r}
manifest <- IlluminaHumanMethylationEPICanno.ilm10b4.hg19::Other %>%
  as.data.frame() %>%
  rownames_to_column(var = "name")
prs_annot <- data.table(top_prs_hits)[manifest, gene := gsub(";.*", "", UCSC_RefGene_Name), on = c(ID = "name")]
ggplot(prs_annot, aes(logFC, -log10(P.Value))) +
  geom_point() +
  geom_point(
    data = subset(prs_annot, adj.P.Val < 0.05 & abs(logFC) > 0.03),
    color = "green",
    mapping = aes(logFC, -log10(P.Value))
  ) +
  geom_hline(
    linetype = "dashed",
    yintercept = min(-log10(prs_annot$P.Value[prs_annot$adj.P.Val < 0.05]))
  ) +
  geom_vline(linetype = "dashed", xintercept = 0.03) +
  geom_vline(linetype = "dashed", xintercept = -0.03) +
  geom_text_repel(
    data = prs_annot %>% filter(abs(logFC) > 0.03 & adj.P.Val < 0.05, P.Value < 1e-10),
    color = "dodgerblue",
    mapping = aes(logFC, -log10(P.Value), label = ifelse(gene != "", gene, ID)),
    size = 3,
    max.overlaps = 20
  ) +
  labs(y = bquote("log"[10] ~ "(P)"), x = quote(Delta ~ beta ~ Methylation)) +
  theme_minimal()
```

# Manhattan plot vs GWAS manhattan plot
```{r}
library(qqman)
cpg_pos <- IlluminaHumanMethylationEPICanno.ilm10b4.hg19::Locations %>%
  as.data.frame() %>%
  rownames_to_column(var = "name")
copy_annot <- prs_annot[cpg_pos, on = c(ID = "name")] %>%
  filter(chr != "chrX|chrY") %>%
  mutate(chr = as.numeric(gsub("chr", "", chr)))
manhattan(copy_annot[, .(SNP = gene, CHR = chr, BP = pos, P = P.Value)],
  annotatePval = max(prs_annot$P.Value[prs_annot$adj.P.Val < 0.05]),
  annotateTop = TRUE,
  genomewideline = min(-log10(prs_annot$P.Value[prs_annot$adj.P.Val < 0.05])),
  suggestiveline = FALSE
)
manhattan(copy_annot[chr == 17, .(SNP = gene, CHR = chr, BP = pos, P = P.Value)],
  annotatePval = max(prs_annot$P.Value[prs_annot$adj.P.Val < 0.05]),
  annotateTop = FALSE,
  genomewideline = min(-log10(prs_annot$P.Value[prs_annot$adj.P.Val < 0.05])),
  suggestiveline = FALSE
)
```
```{r}
GWAS <- fread("~/nalls_PD.QC.gz")
```
```{r}
manhattan(GWAS,
  annotatePval = 5e-8,
  bp = "POS",
  p = "p",
  suggestiveline = FALSE
)
```
