---
title: "MatrixEQTL Data Prep"
output: github_document
---

```{r setup, include=FALSE}
library(tidyverse)
library(data.table)
Sys.setlocale("LC_MESSAGES", "en_US.utf8")
knitr::opts_knit$set(root.dir = "~/prs_ewas_integration/cis_mQTL_analyses")
knitr::opts_chunk$set(echo = TRUE,cache=TRUE,cache.lazy=FALSE,autodep=TRUE)
```

# TERRE

```{r}
terre_ancestry <- fread("~/genotype_qc/TERRE_QC/raw_data.geno.maf.mind.sex_check.het_filter.ibd_filter.eigenvec")
terre_meta <- fread("/home1/NEURO/SHARE_DECIPHER/terre_meta_master.csv")
terre_ctp <- fread("/home1/NEURO/SHARE_DECIPHER/processed_DNAm_data/TERRE/TERRE_funnorm_processed/meta_TERRE_funnorm_robCTP.txt")
terre_meta$IID <- gsub(".*_(PAE.*)","\\1",terre_meta$IID)
terre_meta <- terre_meta %>% right_join(terre_ancestry,by=c("IID"="V2")) %>% left_join(terre_ctp[,.(patient,CTP_PC1, CTP_PC2, CTP_PC3, CTP_PC4, CTP_PC5)],by="patient")
load("/home1/NEURO/SHARE_DECIPHER/processed_DNAm_data/TERRE/TERRE_funnorm_processed/betas_combat.RData")#betas_combat
genotype <- fread("terre_data/raw_data.imputed.r2_30.maf_mismatch.traw")
```

## Match Samples
```{r}
colnames(genotype) <- gsub(".*_(PAE.*)","\\1",colnames(genotype))
colnames(genotype)
setcolorder(genotype,neworder = c(colnames(genotype)[1:6],terre_meta$IID))
betas_combat <- betas_combat[,colnames(betas_combat) %in% terre_meta$patient]
colnames(betas_combat) <- terre_meta$IID[match(colnames(betas_combat),terre_meta$patient)]
betas_combat <- betas_combat[,match(colnames(betas_combat),terre_meta$IID)]

```
```{r}
all(colnames(betas_combat) == terre_meta$IID)
all(colnames(betas_combat) == colnames(genotype)[-c(1:6)])
```
## Compute PCs
```{r}
methy_PC <- prcomp(t(betas_combat), center=T,rank.= 20)
```

## Write Covars

```{r}
cat_vars <- model.matrix(~0+plate, data=terre_meta %>% mutate(plate= as.factor(plate)))
for(i in 0:20){
  if(i == 0){
    covar<- cbind(terre_meta[,c("V3","V4","V5","age","men")])
  }
  else{
    pcs <- methy_PC$x[,1:i]
    covar<- cbind(pcs,terre_meta[,c("V3","V4","V5","age","head_trauma_loc","men")])
  }
  write_delim(t(covar) %>% as.data.frame() %>% rownames_to_column("id"),sprintf("terre_data/covariates_%d_methy_PC.txt",i))
}
write_delim(t(cbind(terre_meta[,c("V3","V4","V5","age","head_trauma_loc","men","CTP_PC1", "CTP_PC2", "CTP_PC3", "CTP_PC4", "CTP_PC5")])) %>% as.data.frame() %>% rownames_to_column("id"),"terre_data/covariates_CTP.txt")

write_delim(t(cbind(terre_meta[terre_meta$men == 1,c("V3","V4","V5","age","head_trauma_loc","CTP_PC1", "CTP_PC2", "CTP_PC3", "CTP_PC4", "CTP_PC5")])) %>% as.data.frame() %>% rownames_to_column("id"),"terre_data/male_covariates_CTP.txt")
write_delim(t(cbind(terre_meta[terre_meta$men == 0,c("V3","V4","V5","age","head_trauma_loc","CTP_PC1", "CTP_PC2", "CTP_PC3", "CTP_PC4", "CTP_PC5")])) %>% as.data.frame() %>% rownames_to_column("id"),"terre_data/female_covariates_CTP.txt")
```

## Write data
```{r}
methy_annot <- fread("~/MethylationEPIC_v-1-0_B4.csv", skip = 7)
```

```{r}
#SNP POS
write_delim(genotype[,.(SNP,CHR=paste0("chr",CHR),POS)],"terre_data/snp_pos.txt")
#SNPs
geno_vars <- colnames(genotype)[-c(1,3:6)]
geno_vars_m <- geno_vars[-(1 + which(terre_meta$men == 0))]
geno_vars_f <- geno_vars[-(1 + which(terre_meta$men == 1))]
fwrite(genotype[,..geno_vars],"terre_data/all_imputed_matrixeQTL.txt",sep = " ",quote = F)
fwrite(genotype[,..geno_vars_f],"terre_data/female_all_imputed_matrixeQTL.txt",sep = " ",quote = F)
fwrite(genotype[,..geno_vars_m],"terre_data/male_all_imputed_matrixeQTL.txt",sep = " ",quote = F)

#Methy POS
fwrite(methy_annot[Name %in% rownames(betas_combat),.(geneid=Name, chr=paste0("chr",CHR),s1=MAPINFO,s2=MAPINFO)], "terre_data/probe_pos.txt",sep = " ",quote=F)
#methy
fwrite(betas_combat %>% as.data.frame() %>%rownames_to_column("cpg"),"terre_data/methylation_combat.txt",sep=" ",quote=F)
fwrite(betas_combat[,terre_meta$men == 1] %>% as.data.frame() %>%rownames_to_column("cpg"),"terre_data/male_methylation_combat.txt",sep=" ",quote=F)
fwrite(betas_combat[,terre_meta$men == 0] %>% as.data.frame() %>%rownames_to_column("cpg"),"terre_data/female_methylation_combat.txt",sep=" ",quote=F)

#SNP POS
write_delim(genotype[CHR==21,.(SNP,CHR=paste0("chr",CHR),POS)],"terre_data/snp_pos_chr21.txt")
#SNPs
geno_vars <- colnames(genotype)[-c(1,3:6)]
write_delim(genotype[CHR==21,..geno_vars],"terre_data/all_imputed_matrixeQTL_chr21.txt")
#Methy POS
write_delim(methy_annot[Name %in% rownames(betas_combat),.(geneid=Name, chr=paste0("chr",CHR),s1=MAPINFO,s2=MAPINFO)][chr=="chr21"], "terre_data/probe_pos_chr21.txt")
chr21_cpg <- methy_annot[Name %in% rownames(betas_combat) & CHR == 21,]$Name
#methy
write_delim(betas_combat %>% as.data.frame() %>%rownames_to_column("cpg") %>% filter(cpg %in% chr21_cpg),"terre_data/methylation_combat_chr21.txt")
```


#Checking number of PCs

```{r}
pc_res <- lapply(0:20,function(i)fread(sprintf("terre_data/cis_all_impute_mQTL_results_%d_methy_PC_chr21.txt",i)))
ctp_res <- fread("terre_data/cis_all_impute_mQTL_results_CTP_chr21.txt")
results_df <- data.frame(PC=0:20,nhit = sapply(pc_res,function(dt)nrow(dt[FDR < 0.05])))
ggplot(results_df,aes(PC,nhit)) + geom_point() +geom_smooth() + geom_hline(yintercept = nrow(ctp_res[FDR < 0.05]),linetype="dashed")+ annotate(geom="text",y=240500,x=10,label="Hits detected with CTP PC") + ggtitle("mQTLs detected on chr21 Vs Methylation PCs included")

results_df <- data.frame(PC=0:20,nhit = sapply(pc_res,function(dt)nrow(dt[FDR < 0.05,.(unique(gene))])))
ggplot(results_df,aes(PC,nhit)) + geom_point() +geom_smooth() + geom_hline(yintercept = nrow(ctp_res[FDR < 0.05,.(unique(gene))]),linetype="dashed") + ggtitle("CpG sites on chr21 with one mQTL Vs Methylation PCs included") + annotate(geom="text",y=4200,x=10,label="Hits detected with CTP PC")
```

