---
title: "DIGPD Data prep"
output: html_document
---

```{r setup, include=FALSE}
library(tidyverse)
library(data.table)
knitr::opts_chunk$set(echo = TRUE)
```

# DIGPD

```{r}
digpd_pcs <- fread("~/genotype_qc/eigenvectors.txt")
digpd_metadata <- fread("/home1/NEURO/SHARE_DECIPHER/digpd_meta_master.csv") %>% select(-PD) %>% mutate(IID = gsub("PAL_[0-9]*_","",IID))
digpd_meta <- digpd_metadata%>% left_join(digpd_pcs,by=c("IID"))
genotype <- fread("~/genotype_qc/DIGPD_merged_post_imputation_QC/raw_data.imputed.r2_90.maf_mismatch.traw")
load("/home1/NEURO/SHARE_DECIPHER/processed_DNAm_data/DIGPD/DIGPDvisit1_betas_combat.RData")#betas_sub

```

## Match Samples
```{r}
colnames(genotype) <- gsub("PAL_[0-9]*_|AOR.*_(AOR.*)","\\1",colnames(genotype))
colnames(genotype)
digpd_meta$patient <- paste0(digpd_meta$patient,".visit",digpd_meta$visite)
digpd_meta <- digpd_meta %>% filter(IID %in% colnames(genotype)& patient %in% colnames(betas_sub))
setcolorder(genotype,neworder = c(colnames(genotype)[1:6],digpd_meta$IID))

betas_sub <- betas_sub[,digpd_meta$patient]
colnames(betas_sub) <- digpd_meta$IID


```
```{r}
all(colnames(betas_sub) == digpd_meta$IID)
all(colnames(betas_sub) == colnames(genotype)[-c(1:6)])
```
## Compute PCs
```{r}
methy_PC <- prcomp(t(betas_sub), center=T,rank.= 20)
```

## Write Covars

```{r}
for(i in 0:20){
  if(i == 0){
    covar<- digpd_meta[,c("U1","U2","U3","Age","men")]
  }
  else{
    pcs <- methy_PC$x[,1:i]
    covar<- cbind(pcs,digpd_meta[,c("U1","U2","U3","Age","men")])
  }
  write_delim(t(covar) %>% as.data.frame() %>% rownames_to_column("id"),sprintf("digpd_data/covariates_%d_methy_PC.txt",i))
}
```

## Write data
```{r}
methy_annot <- fread("~/MethylationEPIC_v-1-0_B4.csv", skip = 7)
```

```{r}
#SNP POS
write_delim(genotype[,.(SNP,CHR=paste0("chr",CHR),POS)],"digpd_data/snp_pos.txt")
#SNPs
geno_vars <- colnames(genotype)[-c(1,3:6)]
fwrite(genotype[,..geno_vars],"digpd_data/all_imputed_matrixeQTL.txt",sep = " ",quote = F)
#Methy POS
fwrite(methy_annot[Name %in% rownames(betas_sub),.(geneid=Name, chr=paste0("chr",CHR),s1=MAPINFO,s2=MAPINFO)], "digpd_data/probe_pos.txt",sep = " ",quote=F)
#methy
fwrite(betas_sub %>% as.data.frame() %>%rownames_to_column("cpg"),"digpd_data/methylation_combat.txt",sep=" ",quote=F)

#SNP POS
write_delim(genotype[CHR==21,.(SNP,CHR=paste0("chr",CHR),POS)],"digpd_data/snp_pos_chr21.txt")
#SNPs
geno_vars <- colnames(genotype)[-c(1,3:6)]
write_delim(genotype[CHR==21,..geno_vars],"digpd_data/all_imputed_matrixeQTL_chr21.txt")
#Methy POS
write_delim(methy_annot[Name %in% rownames(betas_sub),.(geneid=Name, chr=paste0("chr",CHR),s1=MAPINFO,s2=MAPINFO)][chr=="chr21"], "digpd_data/probe_pos_chr21.txt")
chr21_cpg <- methy_annot[Name %in% rownames(betas_sub) & CHR == 21,]$Name
#methy
write_delim(betas_sub %>% as.data.frame() %>%rownames_to_column("cpg") %>% filter(cpg %in% chr21_cpg),"digpd_data/methylation_combat_chr21.txt")
```


#Checking number of PCs

```{r}
pc_res <- lapply(0:20,function(i)fread(sprintf("digpd_data/cis_all_impute_mQTL_results_%d_methy_PCi_chr21.txt",i)))
results_df <- data.frame(PC=0:20,nhit = sapply(pc_res,function(dt)nrow(dt[FDR < 0.05])))
ggplot(results_df,aes(PC,nhit)) + geom_point() +geom_smooth()
```