---
title: "Post mQTL analysis Terre"
output: html_notebook
---

```{r setup, include=FALSE}
library(coloc)
library(data.table)
library(parallel)
library(qqman)
library(qvalue)
library(readxl)
library(Gviz)
library(tidyverse)
knitr::opts_chunk$set(echo = TRUE, cache = TRUE)
knitr::opts_chunk$set(cache = TRUE)
```
## Replication of Cross-sex mQTL
```{r}
library(dbplyr)
library(DBI)
ntests <- 126904049
con <- dbConnect(RSQLite::SQLite(), "terre_and_digpd_cis_all_impute_mQTL_9_methy_PC.db")
mQTL_results <- tbl(con, "terre")
ntests_digpd <- 100363748
mQTL_results_digpd <- tbl(con, "digpd")
```
```{r}
mQTL_results %>%
  filter(FDR < 0.05) %>%
  summarize(total = n(), cpgs = n_distinct(gene))
mQTL_results %>%
  filter(`p-value` < (0.05 / ntests)) %>%
  summarize(total = n(), cpgs = n_distinct(gene))

mQTL_results_digpd %>%
  filter(FDR < 0.05) %>%
  summarize(total = n(), cpgs = n_distinct(gene))
mQTL_results_digpd %>%
  filter(`p-value` < (0.05 / ntests)) %>%
  summarize(total = n(), cpgs = n_distinct(gene))

library(dbplot)
mQTL_results %>% dbplot_histogram(`p-value`)
# qq(mQTL_results$`p-value`)
```

### Checking replication
$\pi_1$ statistics
```{r}
to_check_replication <- mQTL_results_digpd %>%
  inner_join(mQTL_results %>% filter(FDR < 0.05), by = c("SNP", "gene"), copy = TRUE) %>%
  collect() %>% as.data.table()

pi0res <- pi0est(na.omit(as.numeric(to_check_replication$`p-value.x`)), lambda = seq(0, 0.245, 0.005), pi0.method = "bootstrap")
1 - pi0res$pi0
plot(pi0res$pi0.lambda, pi0res$lambda, xlab = bquote(pi[0]), ylab = bquote(lambda))
abline(v = pi0res$pi0)
qvalres <- qvalue_truncp(na.omit(as.numeric(to_check_replication$`p-value.x`)))
1 - qvalres$pi0
abline(v = qvalres$pi0)
pi0est(as.numeric(to_check_replication$`p-value.x`) / 0.25)
```

## Colocalization with parkinson's loci
```{r colocalization, eval=FALSE}
probe_pos <- fread("terre_data/probe_pos.txt")
snp_pos <- fread("terre_data/snp_pos.txt", key = "SNP")
methy <- fread("~/prs_ewas_integration/cis_mQTL_analyses/terre_data/methylation_combat.txt", key = "cpg")
sum_stats <- fread("~/nallsEtAl2019_excluding23andMe_allVariants.tab.gz", key = "SNP")
risk_scores <- read_xlsx("~/pd_risk_score/Supplementary materials/Table S2. Detailed summary statistics on all nominated risk variants, known and novel_.xlsx") %>%
  mutate(CHR = as.numeric(CHR), COUNTED = toupper(`Effect allele`), ALT = toupper(`Other allele`), POS = BP) %>%
  dplyr::filter(!is.na(CHR))
mQTL_results <- fread("terre_data/cis_all_impute_mQTL_results_9_methy_PC.txt.gz", key = "gene")

# Grab CpG sites within 1MB of risk loci
probes_1mb <- function(row) {
  probe_pos[chr == paste0("chr", row$CHR) & s1 >= row$POS - 1e6 & s1 <= row$POS + 1e6]$geneid
}

localize_per_locus <- function(
  i,
  mQTL_results = mQTL_results,
  snp_pos = snp_pos,
  has_methy = TRUE,
  N1 = 245,
  sum_stats=sum_stats,
  risk_scores=risk_scores, # Must contain CHR,POS, SNP, and Locus Number column
  return_summary = TRUE) {
  all_probes <- unlist(probes_1mb(risk_scores[i, ]))
  probe_res <- mclapply(
    1:length(all_probes),
    function(j) {
      snp_set <- mQTL_results[all_probes[j], on = "gene"][!duplicated(SNP)]
      coords <- snp_pos[snp_set$SNP, on = "SNP"][!duplicated(SNP)]
      tmp_stats <- sum_stats[paste0(coords$CHR, ":", coords$POS), on = "SNP"]
      if (has_methy) {
        mqtl_pval <- data.frame(
          beta = snp_set$beta,
          varbeta = snp_set$beta / snp_set$`t-stat`,
          N = N1,
          snp = paste0(coords$CHR, ":", coords$POS),
          sdY = sd(methy[all_probes[j], -c(1), on = "cpg"]),
          type = "quant",
          stringsAsFactors = F
        )
      } else {
        mqtl_pval <- data.frame(
          beta = snp_set$beta,
          N = N1,
          snp = paste0(coords$CHR, ":", coords$POS),
          MAF = tmp_stats$freq,
          pvalues = snp_set$`p-value`,
          type = "quant",
          stringsAsFactors = F
        )
      }
      gwas_pval <- data.frame(
        pvalues = tmp_stats$p,
        N = tmp_stats$N_cases + tmp_stats$N_controls,
        s = tmp_stats$N_cases / (tmp_stats$N_cases + tmp_stats$N_controls),
        snp = tmp_stats$SNP,
        type = "cc",
        MAF = tmp_stats$freq,
        stringsAsFactors = F
      )
      mqtl_pval <- as.list(mqtl_pval[!is.na(gwas_pval$pvalues), ])
      mqtl_pval$N <- unique(mqtl_pval$N)[1]
      mqtl_pval$type <- unique(mqtl_pval$type)[1]
      gwas_pval <- as.list(gwas_pval[!is.na(gwas_pval$pvalues), ])
      gwas_pval$N <- unique(gwas_pval$N)[1]
      gwas_pval$type <- unique(gwas_pval$type)[1]
      print(gwas_pval)
      print(mqtl_pval)
      if (length(gwas_pval$snp) != 0 | length(mqtl_pval$snp != 0)) {
        res <- coloc.abf(mqtl_pval, gwas_pval)
        if (return_summary) {
          return(cbind(t(data.frame(res$summary)), data.frame(probe = all_probes[j], locus = risk_scores$`Locus Number`[i], locus_snp = risk_scores$SNP[i])))
        } else {
          return(cbind(res$results, data.frame(probe = all_probes[j], locus = risk_scores$`Locus Number`[i], locus_snp = risk_scores$SNP[i])))
        }
      } else {
        return(data.frame())
      }
    },
    mc.cores = 4
  )
  return(rbindlist(probe_res))
}
```

```{r}
system.time(res1_pval <- rbindlist(mclapply(1:nrow(risk_scores), function(i) localize_per_locus(i, mQTL_results = mQTL_results, snp_pos = snp_pos, has_methy = FALSE, N1 = 245), mc.cores = 4)))

system.time(res1_SNP <- rbindlist(mclapply(1:nrow(risk_scores), function(i) localize_per_locus(i, mQTL_results = mQTL_results, snp_pos = snp_pos, has_methy = FALSE, N1 = 245, return_summary = FALSE), mc.cores = 4)))
fwrite(res1_SNP, "mqtl_pd_snp_colocalization.txt.gz", sep = "\t", row.names = F, quote = F)
fwrite(res1_pval, "mqtl_pd_snp_colocalization.txt", sep = "\t", row.names = F, quote = F)
```

## Sex stratified colocalization
See [Script](run_colocalization.R).
### Male results at CpG level
```{r}
male_results <- fread("male_mqtl_male_sumstats_pd_snp_colocalization_ph4.txt.gz")
```
Hits at $P(H_4) > 0.9$
```{r}
male_results[PP.H4.abf > 0.9,.(unique_loci = uniqueN(locus), unique_probes= uniqueN(probe))]
male_results[PP.H4.abf > 0.9,.(uniqueN(probe)),by="locus"]
male_results[PP.H4.abf > 0.9,.(uniqueN(locus)),by="probe"]
male_results[PP.H4.abf > 0.9]

```
```{r}
ggplot(male_results_snp[,.(loci_per_snp=uniqueN(locus)), by = "snp"],aes(loci_per_snp))+
  geom_histogram(bins=100)
```

```{r}
male_results_snp <- fread("male_mqtl_male_sumstats_pd_snp_colocalization_per_snp.txt.gz")
```


### Female
```{r}
female_results <- fread("female_mqtl_female_sumstats_pd_snp_colocalization_ph4.txt.gz")
female_results[PP.H4.abf > 0.9,.(unique_loci = uniqueN(locus), unique_probes= uniqueN(probe))]
female_results[PP.H4.abf > 0.9,.(uniqueN(probe)),by="locus"]
female_results[PP.H4.abf > 0.9,.(uniqueN(locus)),by="probe"]
female_results[PP.H4.abf > 0.9]
```

### male vs female overlap

```{r}
res <- list("male"=unique(male_results[PP.H4.abf > 0.9]$probe),"female" = unique(female_results[PP.H4.abf > 0.9]$probe))
sex_shared <- Reduce(function(a,b)a[a %in% b],res)
male <- res$male[!res$male %in% res$female]
female <- res$female[!res$female %in% res$male]
plt <- VennDiagram::venn.diagram(res,filename=NULL,cex = 1,
  cat.cex = 1,
  lwd = 2)
grid.newpage()
grid.draw(plt)
```
## Plot overlap
```{r}
plot_probes_sex <- function(probe_set)
for(cur_probe in probe_set){
  to_plot_male <- male_results_snp[probe == cur_probe]  %>%
    mutate(POS = as.numeric(gsub(".*:","",snp))) %>%
    pivot_longer(c("pvalues.df1","pvalues.df2"),names_to="Var",values_to="Val") %>%
    mutate(Var=recode(Var,pvalues.df1="mQTL",pvalues.df2="GWAS"),Sex="Male")
  to_plot_female <- female_results_snp[probe == cur_probe]  %>%
    mutate(POS = as.numeric(gsub(".*:","",snp))) %>%
    pivot_longer(c("pvalues.df1","pvalues.df2"),names_to="Var",values_to="Val") %>%
    mutate(Var=recode(Var,pvalues.df1="mQTL",pvalues.df2="GWAS"),Sex="Female")
  p<- ggplot(rbind(to_plot_male,to_plot_female),aes(POS,-log10(Val),color=SNP.PP.H4)) +
    geom_point()+
    scale_color_fermenter(palette="Spectral") +
    facet_wrap(Var~Sex)+
    labs(y="-log10 P")+
    ggtitle(cur_probe)
  print(p)
}
```
### Shared
```{r}
plot_probes_sex(sex_shared)
```

### Male
```{r}
plot_probes_sex(male)
```
### Female
```{r}
plot_probes_sex(female)
```
## Mendelian Randomization
```{r}
run_mr_per_cpg <- function(mQTL,cur_gwas){
  mQTL <- mQTL[grepl("rs",V1)]
  cur_gwas <- cur_gwas[mQTL$V1, on = "SNP", nomatch=0][p < 5e-8]
  cur_mQTL <- mQTL[cur_gwas$SNP, on = "V1", nomatch=0]
  if(nrow(cur_mQTL) <= 2){
    return("No GWAS LOCI")
  }else{
    cor_table <- LDmatrix(
      snps=cur_mQTL$V1,
      pop="EUR",
      genome_build = "grch37",
      token = gsub("\n","",readr::read_file("ldlink_token.txt"))
    )
    cor_mat <- as.matrix(cor_table[,-c(1)])
    rownames(cor_mat) <- as.character(cor_table$RS_number)
    cur_mQTL <- cur_mQTL[rownames(cor_mat), on="V1"]
    cur_gwas <- cur_gwas[rownames(cor_mat), on="SNP"]
    
    EggerObject <- try(
      mr_egger(
        mr_input(
          bx = cur_mQTL$V3, 
          bxse = cur_mQTL$V3/cur_mQTL$V4 ,
          by = cur_gwas$b, 
          byse = cur_gwas$se,
          correlation = cor_mat
        ),
        correl = TRUE,
        distribution = "t-dist",
        alpha = 0.05
      ),
      silent =  TRUE
    )
    if(class(EggerObject) == 'try-error'){
      return(glue("{nrow(cur_mQTL)} mQTL, MR not fit"))
    }else{
      return(EggerObject)
    }
  }
}
```

