---
title: "Post mQTL analysis Terre"
output: html_notebook
---

```{r setup, include=FALSE}
library(coloc)
library(data.table)
library(parallel)
library(qqman)
library(qvalue)
library(readxl)
library(Gviz)
library(tidyverse)
knitr::opts_chunk$set(echo = TRUE, cache = TRUE)
knitr::opts_chunk$set(cache = TRUE)
```
## Replication of Cross-sex mQTL
```{r}
library(dbplyr)
library(DBI)
ntests <- 126904049
con <- dbConnect(RSQLite::SQLite(), "terre_and_digpd_cis_all_impute_mQTL_9_methy_PC.db")
mQTL_results <- tbl(con, "terre")
ntests_digpd <- 100363748
mQTL_results_digpd <- tbl(con, "digpd")
```
```{r}
mQTL_results %>%
  filter(FDR < 0.05) %>%
  summarize(total = n(), cpgs = n_distinct(gene))
mQTL_results %>%
  filter(`p-value` < (0.05 / ntests)) %>%
  summarize(total = n(), cpgs = n_distinct(gene))

mQTL_results_digpd %>%
  filter(FDR < 0.05) %>%
  summarize(total = n(), cpgs = n_distinct(gene))
mQTL_results_digpd %>%
  filter(`p-value` < (0.05 / ntests)) %>%
  summarize(total = n(), cpgs = n_distinct(gene))

library(dbplot)
mQTL_results %>% dbplot_histogram(`p-value`)
# qq(mQTL_results$`p-value`)
```

### Checking replication
$\pi_1$ statistics
```{r}
to_check_replication <- mQTL_results_digpd %>%
  inner_join(mQTL_results %>% filter(FDR < 0.05), by = c("SNP", "gene"), copy = TRUE) %>%
  collect() %>% as.data.table()

pi0res <- pi0est(na.omit(as.numeric(to_check_replication$`p-value.x`)), lambda = seq(0, 0.245, 0.005), pi0.method = "bootstrap")
1 - pi0res$pi0
plot(pi0res$pi0.lambda, pi0res$lambda, xlab = bquote(pi[0]), ylab = bquote(lambda))
abline(v = pi0res$pi0)
qvalres <- qvalue_truncp(na.omit(as.numeric(to_check_replication$`p-value.x`)))
1 - qvalres$pi0
abline(v = qvalres$pi0)
pi0est(as.numeric(to_check_replication$`p-value.x`) / 0.25)
```

# Colocalization with parkinson's loci and overlap with ewas hits
```{r}
load("../prs_analyses/prs_nalls_cross_w_sex_stratified.RData")
manifest <- IlluminaHumanMethylationEPICanno.ilm10b4.hg19::Other %>%
  as.data.frame() %>%
  rownames_to_column(var = "name")
```

## cross-sex GWAS
```{r}
cross_coloc <- fread("cross_mqtl_cross_sumstats_pd_snp_colocalization_ph4.txt.gz")
male_coloc <- fread("male_mqtl_cross_sumstats_pd_snp_colocalization_ph4.txt.gz")
female_coloc <- fread("female_mqtl_cross_sumstats_pd_snp_colocalization_ph4.txt.gz")
```
### Cross-sex mQTL
```{r}
sig_bonf_cross <- top_prs_hits %>% filter(P.Value < (0.05 / n()))
cross_coloc %>%
  filter(PP.H4.abf > 0.9,probe %in% sig_bonf_cross$ID) %>%
  group_by(probe) %>%
  slice(which.max(PP.H4.abf)) %>%
  left_join(manifest,by = c("probe"="name"))

cross_coloc %>%
  filter(PP.H4.abf < 0.9,probe %in% sig_bonf_cross$ID) %>%
  group_by(probe) %>%
  slice(which.max(PP.H4.abf)) %>%
  left_join(manifest,by = c("probe"="name"))

cross_coloc %>%
  filter(PP.H4.abf > 0.9,!probe %in% sig_bonf_cross$ID) %>%
  group_by(probe) %>%
  slice(which.max(PP.H4.abf)) %>%
  left_join(manifest,by = c("probe"="name"))
```
### Male mQTL
```{r}
sig_bonf_male <- top_male_prs_hits %>% filter(P.Value < (0.05 / n()))
male_coloc %>%
  filter(PP.H4.abf > 0.9,probe %in% sig_bonf_male$ID) %>%
  group_by(probe) %>%
  slice(which.max(PP.H4.abf)) %>%
  left_join(manifest,by = c("probe"="name"))

male_coloc %>%
  filter(PP.H4.abf < 0.9,probe %in% sig_bonf_male$ID) %>%
  group_by(probe) %>%
  slice(which.max(PP.H4.abf)) %>%
  left_join(manifest,by = c("probe"="name"))

male_coloc %>%
  filter(PP.H4.abf > 0.9,!probe %in% sig_bonf_male$ID) %>%
  group_by(probe) %>%
  slice(which.max(PP.H4.abf)) %>%
  left_join(manifest,by = c("probe"="name"))
```

### Female mQTL
```{r}
sig_bonf_female <- top_female_prs_hits %>% filter(P.Value < (0.05 / n()))
female_coloc %>%
  filter(PP.H4.abf > 0.9,probe %in% sig_bonf_female$ID) %>%
  group_by(probe) %>%
  slice(which.max(PP.H4.abf)) %>%
  left_join(manifest,by = c("probe"="name"))

female_coloc %>%
  filter(PP.H4.abf < 0.9,probe %in% sig_bonf_female$ID) %>%
  group_by(probe) %>%
  slice(which.max(PP.H4.abf)) %>%
  left_join(manifest,by = c("probe"="name"))

female_coloc %>%
  filter(PP.H4.abf > 0.9,!probe %in% sig_bonf_female$ID) %>%
  group_by(probe) %>%
  slice(which.max(PP.H4.abf)) %>%
  left_join(manifest,by = c("probe"="name"))
```

## Sex stratified GWAS colocalization
```{r}
load("../prs_analyses/prs_blau_sex_stratified.RData")
male_stratified_coloc <- fread("male_mqtl_male_sumstats_pd_snp_colocalization_ph4.txt.gz")
female_stratified_coloc <- fread("female_mqtl_female_sumstats_pd_snp_colocalization_ph4.txt.gz")
```

See [Script](run_colocalization.R).
### Male results at CpG level
```{r}
sig_FDR_male <- top_male_ewas %>% filter(adj.P.Val < 0.25)
male_stratified_coloc %>%
  filter(PP.H4.abf > 0.9,probe %in% sig_FDR_male$ID) %>%
  group_by(probe) %>%
  slice(which.max(PP.H4.abf)) %>%
  left_join(manifest,by = c("probe"="name"))

male_stratified_coloc %>%
  filter(PP.H4.abf < 0.9,probe %in% sig_FDR_male$ID) %>%
  group_by(probe) %>%
  slice(which.max(PP.H4.abf)) %>%
  left_join(manifest,by = c("probe"="name"))

male_stratified_coloc %>%
  filter(PP.H4.abf > 0.9,!probe %in% sig_FDR_male$ID) %>%
  group_by(probe) %>%
  slice(which.max(PP.H4.abf)) %>%
  left_join(manifest,by = c("probe"="name"))
```

```{r}
sig_FDR_female <- top_female_ewas %>% filter(adj.P.Val < 0.25)
female_stratified_coloc %>%
  filter(PP.H4.abf > 0.9,probe %in% sig_FDR_female$ID) %>%
  group_by(probe) %>%
  slice(which.max(PP.H4.abf)) %>%
  left_join(manifest,by = c("probe"="name"))

female_stratified_coloc %>%
  filter(PP.H4.abf < 0.9,probe %in% sig_FDR_female$ID) %>%
  group_by(probe) %>%
  slice(which.max(PP.H4.abf)) %>%
  left_join(manifest,by = c("probe"="name"))

female_stratified_coloc %>%
  filter(PP.H4.abf > 0.9,!probe %in% sig_FDR_female$ID) %>%
  group_by(probe) %>%
  slice(which.max(PP.H4.abf)) %>%
  left_join(manifest,by = c("probe"="name"))
```

### male vs female overlap

```{r}
res <- list("male"=unique(male_results[PP.H4.abf > 0.9]$probe),"female" = unique(female_results[PP.H4.abf > 0.9]$probe))
sex_shared <- Reduce(function(a,b)a[a %in% b],res)
male <- res$male[!res$male %in% res$female]
female <- res$female[!res$female %in% res$male]
plt <- VennDiagram::venn.diagram(res,filename=NULL,cex = 1,
  cat.cex = 1,
  lwd = 2)
grid.newpage()
grid.draw(plt)
```
## Plot overlap
```{r}
plot_probes_sex <- function(probe_set)
for(cur_probe in probe_set){
  to_plot_male <- male_results_snp[probe == cur_probe]  %>%
    mutate(POS = as.numeric(gsub(".*:","",snp))) %>%
    pivot_longer(c("pvalues.df1","pvalues.df2"),names_to="Var",values_to="Val") %>%
    mutate(Var=recode(Var,pvalues.df1="mQTL",pvalues.df2="GWAS"),Sex="Male")
  to_plot_female <- female_results_snp[probe == cur_probe]  %>%
    mutate(POS = as.numeric(gsub(".*:","",snp))) %>%
    pivot_longer(c("pvalues.df1","pvalues.df2"),names_to="Var",values_to="Val") %>%
    mutate(Var=recode(Var,pvalues.df1="mQTL",pvalues.df2="GWAS"),Sex="Female")
  p<- ggplot(rbind(to_plot_male,to_plot_female),aes(POS,-log10(Val),color=SNP.PP.H4)) +
    geom_point()+
    scale_color_fermenter(palette="Spectral") +
    facet_wrap(Var~Sex)+
    labs(y="-log10 P")+
    ggtitle(cur_probe)
  print(p)
}
```
### Shared
```{r}
plot_probes_sex(sex_shared)
```

### Male
```{r}
plot_probes_sex(male)
```
### Female
```{r}
plot_probes_sex(female)
```
###
## Mendelian Randomization
```{r}
run_mr_per_cpg <- function(mQTL,cur_gwas){
  mQTL <- mQTL[grepl("rs",V1)]
  cur_gwas <- cur_gwas[mQTL$V1, on = "SNP", nomatch=0][p < 5e-8]
  cur_mQTL <- mQTL[cur_gwas$SNP, on = "V1", nomatch=0]
  if(nrow(cur_mQTL) <= 2){
    return("No GWAS LOCI")
  }else{
    cor_table <- LDmatrix(
      snps=cur_mQTL$V1,
      pop="EUR",
      genome_build = "grch37",
      token = gsub("\n","",readr::read_file("ldlink_token.txt"))
    )
    cor_mat <- as.matrix(cor_table[,-c(1)])
    rownames(cor_mat) <- as.character(cor_table$RS_number)
    cur_mQTL <- cur_mQTL[rownames(cor_mat), on="V1"]
    cur_gwas <- cur_gwas[rownames(cor_mat), on="SNP"]

    EggerObject <- try(
      mr_egger(
        mr_input(
          bx = cur_mQTL$V3,
          bxse = cur_mQTL$V3/cur_mQTL$V4 ,
          by = cur_gwas$b,
          byse = cur_gwas$se,
          correlation = cor_mat
        ),
        correl = TRUE,
        distribution = "t-dist",
        alpha = 0.05
      ),
      silent =  TRUE
    )
    if(class(EggerObject) == 'try-error'){
      return(glue("{nrow(cur_mQTL)} mQTL, MR not fit"))
    }else{
      return(EggerObject)
    }
  }
}
```
